{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import difflib\n",
    "from sqlalchemy import create_engine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "gt_df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "gmmb_df = pd.read_csv('data/gmmb_tracker.csv', header=1)\n",
    "ancestry_df = pd.read_csv('data/FEB_2024_cohort_ancestry_phenotype.csv', header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# def find_best_matches(list1, list2, cutoff=0.6):\n",
    "#     matches = {}\n",
    "#     for item in list1:\n",
    "#         match = difflib.get_close_matches(item, list2, n=1, cutoff=cutoff)\n",
    "#         if match:\n",
    "#             matches[item] = match[0]\n",
    "#         else:\n",
    "#             matches[item] = None\n",
    "#     return matches\n",
    "\n",
    "# best_matches = find_best_matches(gt_df.loc[~gt_df['Study Code'].isna(), 'Study Code'], gmmb_df['Cohort Name'])\n",
    "\n",
    "# for k, v in best_matches.items():\n",
    "#     print(f\"{k}: {v}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>New since last data release?</th>\n",
       "      <th>DTI alias for Cohort Name</th>\n",
       "      <th>Name on Ancestry Phenotype Backup</th>\n",
       "      <th>Cohort Name</th>\n",
       "      <th>Members</th>\n",
       "      <th>Institution</th>\n",
       "      <th>City</th>\n",
       "      <th>Country (Location)</th>\n",
       "      <th>Shared to\\n AMP-PD</th>\n",
       "      <th>Expected Samples for\\nPD CASES</th>\n",
       "      <th>...</th>\n",
       "      <th>Prodromal - EUR - General European Ancestry</th>\n",
       "      <th>Prodromal - FIN - Finnish Population Isolate</th>\n",
       "      <th>Prodromal - AMR - Latino and Indigenous People of the Americas</th>\n",
       "      <th>Prodromal - SAS - South Asian Ancestry</th>\n",
       "      <th>Prodromal - CAS - Central Asian Ancestry</th>\n",
       "      <th>Prodromal - MDE - Middle Eastern Ancestry</th>\n",
       "      <th>Prodromal CAH - Complex Admixture History</th>\n",
       "      <th>Completed Samples for PRODROMAL</th>\n",
       "      <th>Total Non-PD Samples Competed</th>\n",
       "      <th>Total Completed Samples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>POM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Radboud University Medical Center</td>\n",
       "      <td>Nejmegen</td>\n",
       "      <td>Netherlands</td>\n",
       "      <td>No</td>\n",
       "      <td>520</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ProPark</td>\n",
       "      <td>NaN</td>\n",
       "      <td>University Medical Center Utrecht.</td>\n",
       "      <td>Utrecht</td>\n",
       "      <td>Netherlands</td>\n",
       "      <td>No</td>\n",
       "      <td>1,250</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 62 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     New since last data release?  DTI alias for Cohort Name  \\\n",
       "156                           NaN                        NaN   \n",
       "187                           NaN                        NaN   \n",
       "\n",
       "    Name on Ancestry Phenotype Backup Cohort Name Members  \\\n",
       "156                               NaN         POM     NaN   \n",
       "187                               NaN     ProPark     NaN   \n",
       "\n",
       "                            Institution      City Country (Location)  \\\n",
       "156   Radboud University Medical Center  Nejmegen        Netherlands   \n",
       "187  University Medical Center Utrecht.   Utrecht        Netherlands   \n",
       "\n",
       "    Shared to\\n AMP-PD Expected Samples for\\nPD CASES  ...  \\\n",
       "156                 No                            520  ...   \n",
       "187                 No                          1,250  ...   \n",
       "\n",
       "    Prodromal - EUR - General European Ancestry  \\\n",
       "156                                           0   \n",
       "187                                           0   \n",
       "\n",
       "    Prodromal - FIN - Finnish Population Isolate  \\\n",
       "156                                            0   \n",
       "187                                            0   \n",
       "\n",
       "     Prodromal - AMR - Latino and Indigenous People of the Americas  \\\n",
       "156                                                  0                \n",
       "187                                                  0                \n",
       "\n",
       "     Prodromal - SAS - South Asian Ancestry  \\\n",
       "156                                       0   \n",
       "187                                       0   \n",
       "\n",
       "     Prodromal - CAS - Central Asian Ancestry  \\\n",
       "156                                         0   \n",
       "187                                         0   \n",
       "\n",
       "     Prodromal - MDE - Middle Eastern Ancestry  \\\n",
       "156                                          0   \n",
       "187                                          0   \n",
       "\n",
       "    Prodromal CAH - Complex Admixture History  \\\n",
       "156                                         0   \n",
       "187                                         0   \n",
       "\n",
       "     Completed Samples for PRODROMAL  Total Non-PD Samples Competed  \\\n",
       "156                                0                              0   \n",
       "187                                0                              0   \n",
       "\n",
       "     Total Completed Samples  \n",
       "156                        0  \n",
       "187                        0  \n",
       "\n",
       "[2 rows x 62 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gmmb_df.loc[gmmb_df['Country (Location)'].str.contains('Netherlands')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Date', 'Updater', 'Study Code', 'Individual Study Name', 'Institution',\n",
       "       'Monogenic / Complex / Mixed',\n",
       "       'if Mixed: estimated proportion in Monogenic', 'City',\n",
       "       'Geographic locality [free text]', 'PI', 'PI email', 'Contact',\n",
       "       'Contact email', 'cohort call status (pre-registration)',\n",
       "       'Data overview approved by Cohort Integration group  [0 = no, 1 = yes]',\n",
       "       'Compliance team approved [0 = no, 1 = yes]',\n",
       "       'Compliance Introductory Call Status', 'Collaboration Agreement Status',\n",
       "       'Material/Data Transfer Agreement Status',\n",
       "       'All necessary agreements executed [0 = no, 1 = yes]',\n",
       "       'Contracting Notes',\n",
       "       'Minimal CDE info recieved and shared to CD-DAWG [0 = no, 1 = yes]',\n",
       "       'Full data [0 = only summary stats, 1 = genotypes sharable]',\n",
       "       'GDPR [0 = no European restriction, 1 = limited by GDPR]',\n",
       "       'Other concerns [Free text]', 'Site [NIH, Tubingen/DZNE, Fulgent, UCL]',\n",
       "       'Contact.1', 'Contact Email', 'DNA samples recieved at site [N]',\n",
       "       'DNA samples attempted [N]', 'Total samples passing basic QC [N]',\n",
       "       'Failed call rates [N < 95%]',\n",
       "       'Failed sex check [N with contradictory reported versus genetic sex]',\n",
       "       'Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]',\n",
       "       'Duplicate samples [N, IBD estiamtes > 95%]',\n",
       "       'Removed relatives [N samples related to one or more samples at IBD > 12.5%]',\n",
       "       'African (AFR)', 'African admixed (AAC)', 'Ashkenzai (AJ)',\n",
       "       'East Asian (EAS)', 'European (EUR)', 'Finnish (FIN)', 'LatinX (AMR)',\n",
       "       'South Asian (SAS)', 'Central Asian (CAS)', 'Middle Eastern (MDE)',\n",
       "       'Complex Admixture (CAH)', 'Total',\n",
       "       'Genotyping complete for the foreseeable future [0 = no, 1 = yes]',\n",
       "       'Imputation panels(s) [Free text]',\n",
       "       'Imputation complete for current samples [0 = no, 1 = yes]',\n",
       "       'Raw data returned to PI [0 = no, 1 =yes]',\n",
       "       'Shared to AMP-PD [0 = no, 1 = yes]',\n",
       "       'Analysts responsible for QC, imputation and transfer [free text]',\n",
       "       'Cohort type [PD/Atypical/Prodromal/Controls only]',\n",
       "       'Predicted # PD cases by 2028', 'Predicted # controls by 2028',\n",
       "       'Predicted # PSP by 2028', 'Predicted # MSA by 2028',\n",
       "       'Predicted # DLB by 2028', 'Predicted # CBS by 2028',\n",
       "       'Predicted # Prodromal by 2028',\n",
       "       'Predicted # Other atypical Parkinsonism cases by 2028',\n",
       "       'Cohort to be WGSd\\nYes = 1\\nNo = 0', 'WGS important notes',\n",
       "       'Sequenced samples [N] ', 'Samples passed WGS QC [N]',\n",
       "       'Shared to AMP-PD [0 = no, 1 = yes].1', 'Notes [free text]'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gt_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genotracker\n",
      "   Study Code                              Individual Study Name  \\\n",
      "58  BENHAMIDA  Clinicogenetic study of Parkinsonâ€™s disease in...   \n",
      "\n",
      "                                         Institution   City  \\\n",
      "58  National Institute Mongi Ben Hamida of Neurology  Tunis   \n",
      "\n",
      "   Geographic locality [free text]  \n",
      "58                         Tunisia  \n",
      "############################################################\n",
      "\n",
      "gmmb\n",
      "                              Cohort Name  \\\n",
      "35  Clinicogenetic Study of PD in Tunisia   \n",
      "\n",
      "                                         Institution   City Country (Location)  \n",
      "35  National Institute of Neurology Mongi-Ben Hamida  Tunis            Tunisia  \n",
      "############################################################\n",
      "\n"
     ]
    }
   ],
   "source": [
    "gt_fields = ['Study Code', 'Individual Study Name', 'Institution', 'City', 'Geographic locality [free text]']\n",
    "gt_fields_df = gt_df.loc[gt_df['Study Code']=='BENHAMIDA', gt_fields]\n",
    "\n",
    "print('genotracker')\n",
    "print(gt_fields_df)\n",
    "print('##'*30)\n",
    "print()\n",
    "\n",
    "gmmb_fields = ['Cohort Name', 'Institution', 'City', 'Country (Location)']\n",
    "gmmb_fields_df = gmmb_df.loc[gmmb_df['Country (Location)'].str.contains('Tunisia'), gmmb_fields]\n",
    "\n",
    "print('gmmb')\n",
    "print(gmmb_fields_df)\n",
    "print('##'*30)\n",
    "print()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Row 0\n",
      " Study Code: BCM\n",
      " Individual Study Name: Baylor College of Medicine (BCM)\n",
      " Cohort Name: Baylor College of Medicine (BCM)\n",
      "Row 1\n",
      " Study Code: CORIELL\n",
      " Individual Study Name: CORIELL\n",
      " Cohort Name: CORIELL\n",
      "Row 2\n",
      " Study Code: PDGNRTN\n",
      " Individual Study Name: PDGENEration\n",
      " Cohort Name: PDGENEration\n",
      "Row 3\n",
      " Study Code: PROSPOS\n",
      " Individual Study Name: PROSPOS\n",
      " Cohort Name: PROSPOS\n",
      "Row 4\n",
      " Study Code: NZP3\n",
      " Individual Study Name: New Zealand Parkinson's Programme\n",
      " Cohort Name: New Zealand Parkinson's Programme\n",
      "Row 5\n",
      " Study Code: BADGEPD\n",
      " Individual Study Name: BADGE-PD\n",
      " Cohort Name: BADGE-PD\n",
      "Row 6\n",
      " Study Code: CAMPAIGN\n",
      " Individual Study Name: BADGE-PD\n",
      " Cohort Name: BADGE-PD\n",
      "Row 7\n",
      " Study Code: DIGPD\n",
      " Individual Study Name: DIGPD\n",
      " Cohort Name: DIGPD\n",
      "Row 8\n",
      " Study Code: ICEBERG\n",
      " Individual Study Name: ICEBERG\n",
      " Cohort Name: ICEBERG\n",
      "Row 9\n",
      " Study Code: IDCME\n",
      " Individual Study Name: NGC\n",
      " Cohort Name: NGC\n",
      "Row 10\n",
      " Study Code: PICNICS\n",
      " Individual Study Name: PICNICS\n",
      " Cohort Name: PICNICS\n",
      "Row 11\n",
      " Study Code: RBM\n",
      " Individual Study Name: RBM\n",
      " Cohort Name: RBM\n",
      "Row 12\n",
      " Study Code: POM\n",
      " Individual Study Name: POM\n",
      " Cohort Name: POM\n",
      "Row 13\n",
      " Study Code: nan\n",
      " Individual Study Name: Sudden Death Cohort\n",
      " Cohort Name: Sudden Death Cohort\n",
      "Row 14\n",
      " Study Code: LCC\n",
      " Individual Study Name: LCC\n",
      " Cohort Name: LCC\n",
      "Row 15\n",
      " Study Code: GPARKGR\n",
      " Individual Study Name: GenePARKGR\n",
      " Cohort Name: GenePARKGR\n",
      "Row 16\n",
      " Study Code: PROPARK\n",
      " Individual Study Name: ProPark\n",
      " Cohort Name: ProPark\n",
      "Row 17\n",
      " Study Code: BIOFIND\n",
      " Individual Study Name: BioFIND\n",
      " Cohort Name: BioFIND\n",
      "Row 18\n",
      " Study Code: CALYPSO\n",
      " Individual Study Name: CALYPSO\n",
      " Cohort Name: CALYPSO\n",
      "Row 19\n",
      " Study Code: PFP\n",
      " Individual Study Name: PFP\n",
      " Cohort Name: PFP\n",
      "Row 20\n",
      " Study Code: PROBAND\n",
      " Individual Study Name: PROBAND\n",
      " Cohort Name: PROBAND\n",
      "Row 21\n",
      " Study Code: VIPD\n",
      " Individual Study Name: VIPD\n",
      " Cohort Name: VIPD\n",
      "Row 22\n",
      " Study Code: CANDAS\n",
      " Individual Study Name: CANDAS\n",
      " Cohort Name: CANDAS\n",
      "Row 23\n",
      " Study Code: QPP\n",
      " Individual Study Name: QPP\n",
      " Cohort Name: QPP\n",
      "Row 24\n",
      " Study Code: MDGAP-MBB\n",
      " Individual Study Name: Manchester Brain Bank\n",
      " Cohort Name: Manchester Brain Bank\n",
      "Row 25\n",
      " Study Code: GENEFINDER\n",
      " Individual Study Name: GeneFinder\n",
      " Cohort Name: GeneFinder\n",
      "Row 26\n",
      " Study Code: OPDC\n",
      " Individual Study Name: OPDC Discovery Cohort\n",
      " Cohort Name: OPDC Discovery Cohort\n",
      "Row 27\n",
      " Study Code: PROATYP\n",
      " Individual Study Name: proATYP\n",
      " Cohort Name: proATYP\n",
      "Row 28\n",
      " Study Code: MDGAP-IBB\n",
      " Individual Study Name: PUKBB\n",
      " Cohort Name: PUKBB\n",
      "Row 29\n",
      " Study Code: PDSGR\n",
      " Individual Study Name: PDS\n",
      " Cohort Name: PDS\n",
      "Row 30\n",
      " Study Code: PREDICTPD\n",
      " Individual Study Name: PREDICT-PD\n",
      " Cohort Name: PREDICT-PD\n",
      "Row 31\n",
      " Study Code: MPBC\n",
      " Individual Study Name: MPBC\n",
      " Cohort Name: MPBC\n",
      "Row 32\n",
      " Study Code: SYDPDRC\n",
      " Individual Study Name: Sydney PDRC\n",
      " Cohort Name: Sydney PDRC\n",
      "Row 33\n",
      " Study Code: IPDGCEA-JKU\n",
      " Individual Study Name: JPDS\n",
      " Cohort Name: JPDS\n",
      "Row 34\n",
      " Study Code: GARVAN\n",
      " Individual Study Name: Genetic analysis in inherited neurological disorders\n",
      " Cohort Name: Genetic analysis in inherited neurological disorders\n",
      "Row 35\n",
      " Study Code: PDMDCB\n",
      " Individual Study Name: Parkinson Disease and Movement Disorders Center Biorepository\n",
      " Cohort Name: Parkinson Disease and Movement Disorders Center Biorepository\n",
      "Row 36\n",
      " Study Code: TUEPAC\n",
      " Individual Study Name: Tuepac + TREND\n",
      " Cohort Name: Tuepac + TREND\n",
      "Row 37\n",
      " Study Code: LUXPARK\n",
      " Individual Study Name: LUX-Park\n",
      " Cohort Name: LUX-Park\n",
      "Row 38\n",
      " Study Code: PDGEN\n",
      " Individual Study Name: PD-GEN and PARKNET\n",
      " Cohort Name: PD-GEN and PARKNET\n",
      "Row 39\n",
      " Study Code: DATATOP\n",
      " Individual Study Name: DATATOP\n",
      " Cohort Name: DATATOP\n",
      "Row 40\n",
      " Study Code: SZTE\n",
      " Individual Study Name: SZTE\n",
      " Cohort Name: SZTE\n",
      "Row 41\n",
      " Study Code: NETPDLS1\n",
      " Individual Study Name: NET-PD\n",
      " Cohort Name: NET-PD\n",
      "Row 42\n",
      " Study Code: DENOPA\n",
      " Individual Study Name: DeNoPa\n",
      " Cohort Name: DeNoPa\n",
      "Row 43\n",
      " Study Code: BBDP\n",
      " Individual Study Name: Arizona Brain Bank Brain and body donation programme (BBDP)\n",
      " Cohort Name: Arizona Brain Bank Brain and body donation programme (BBDP)\n",
      "Row 44\n",
      " Study Code: PDCARE\n",
      " Individual Study Name: PDCARE\n",
      " Cohort Name: PDCARE\n",
      "Row 45\n",
      " Study Code: RAPSODI\n",
      " Individual Study Name: RAPSODI\n",
      " Cohort Name: RAPSODI\n",
      "Row 46\n",
      " Study Code: YOPDFL\n",
      " Individual Study Name: YOPDFL (Young Onset Parkinson's Disease)\n",
      " Cohort Name: YOPDFL (Young Onset Parkinson's Disease)\n",
      "Row 47\n",
      " Study Code: TONIC-PERRON\n",
      " Individual Study Name: TONiC PD\n",
      " Cohort Name: TONiC PD\n",
      "Row 48\n",
      " Study Code: JGUM\n",
      " Individual Study Name: JGUM\n",
      " Cohort Name: JGUM\n",
      "Row 49\n",
      " Study Code: PATRAS\n",
      " Individual Study Name: PATRAS\n",
      " Cohort Name: PATRAS\n",
      "Row 50\n",
      " Study Code: CHCHPD\n",
      " Individual Study Name: Christchurch Longitudinal Parkinson's Study\n",
      " Cohort Name: Christchurch Longitudinal Parkinson's Study\n",
      "Row 51\n",
      " Study Code: QMAP\n",
      " Individual Study Name: Quantitative MRI for Anatomical Phenotyping in Parkinson's Disease\n",
      " Cohort Name: Quantitative MRI for Anatomical Phenotyping in Parkinson's Disease\n",
      "Row 52\n",
      " Study Code: ANANEOS\n",
      " Individual Study Name: ANANEOS Study\n",
      " Cohort Name: ANANEOS Study\n",
      "Row 53\n",
      " Study Code: VETPD\n",
      " Individual Study Name: VET-PD\n",
      " Cohort Name: VET-PD\n",
      "Row 54\n",
      " Study Code: SINAI\n",
      " Individual Study Name: Mount Sinai Cohort\n",
      " Cohort Name: Mount Sinai Cohort\n",
      "Row 55\n",
      " Study Code: DODONA\n",
      " Individual Study Name: The DodoNa Project: DNA Predictions to Improve Neurological Health\n",
      " Cohort Name: The DodoNa Project: DNA Predictions to Improve Neurological Health\n",
      "Row 56\n",
      " Study Code: HAWAIIPD\n",
      " Individual Study Name: HAWAII PD\n",
      " Cohort Name: HAWAII PD\n",
      "Row 57\n",
      " Study Code: AAPDGC\n",
      " Individual Study Name: AfrAbia-PD Genomic Consortium (AA-PDGC)\n",
      " Cohort Name: AfrAbia-PD Genomic Consortium (AA-PDGC)\n",
      "Row 58\n",
      " Study Code: REGARDS\n",
      " Individual Study Name: REasons for Geographic and Racial Differences in Stroke (REGARDS) Project\n",
      " Cohort Name: REasons for Geographic and Racial Differences in Stroke (REGARDS) Project\n",
      "Row 59\n",
      " Study Code: NZPEGS\n",
      " Individual Study Name: New Zealand Parkinson's Environment and Genes Study\n",
      " Cohort Name: New Zealand Parkinson's Environment and Genes Study\n",
      "Row 60\n",
      " Study Code: VAIBB\n",
      " Individual Study Name: Van Andel Institute Brain Bank\n",
      " Cohort Name: Van Andel Institute Brain Bank\n",
      "Row 61\n",
      " Study Code: GPS\n",
      " Individual Study Name: GenePool Study\n",
      " Cohort Name: GenePool Study\n",
      "Row 62\n",
      " Study Code: NGP\n",
      " Individual Study Name: The Neuro Genomics Partnership\n",
      " Cohort Name: The Neuro Genomics Partnership\n",
      "Row 63\n",
      " Study Code: nan\n",
      " Individual Study Name: Kyrgyzstan PD\n",
      " Cohort Name: Kyrgyzstan PD\n"
     ]
    }
   ],
   "source": [
    "gt_name_merge = gt_df[['Study Code','Individual Study Name']].merge(gmmb_df[['Cohort Name']], left_on='Individual Study Name', right_on='Cohort Name')\n",
    "for index, row in gt_name_merge.iterrows():\n",
    "    print(f'Row {index}')\n",
    "    for col_name, value in row.items():\n",
    "        print(f\" {col_name}: {value}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_mapping = {\n",
    "    \"BCM\": \"Baylor College of Medicine (BCM)\",\n",
    "    \"CORIELL\": \"CORIELL\",\n",
    "    \"PDGNRTN\": \"PDGENEration\",\n",
    "    \"UMD\": \"BCM-UMD\",\n",
    "    \"PROSPOS\": \"PROSPOS\",\n",
    "    \"SP-SEV\": \"Sp_SEV - HUVR/CSIC/Universidad de Sevill\",\n",
    "    \"COPN\": \"C-OPN\",\n",
    "    \"NZP3\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"UPENNU19\": \"UPENN U19 Cohort\",\n",
    "    \"BADGEPD\": \"BADGE-PD\",\n",
    "    \"CAMPAIGN\": \"CamPaIGN\",\n",
    "    \"DIGPD\": \"DIGPD\",\n",
    "    \"ICEBERG\": \"ICEBERG\",\n",
    "    \"IDCME\": \"NGC\",\n",
    "    \"MIND\": \"MIND Initiative\",\n",
    "    \"NSPARK\": \"NS PARK\",\n",
    "    \"PICNICS\": \"PICNICS\",\n",
    "    \"PREDISTIM\": \"PREDICT-PD\",\n",
    "    \"QPN\": \"Quebec Parkinson's Network\",\n",
    "    \"RBM\": \"RBM\",\n",
    "    \"SP-BCN3\": \"Sp_BCN3 - Sant Joan de DÃ©u\",\n",
    "    \"SP-OVI\": \"Sp_OVI - University of Oviedo\",\n",
    "    \"MDGAP-KINGS\": \"MD GAP - King's College London\",\n",
    "    \"MDGAP-QSBB\": \"MD GAP - QSBB - University College London\",\n",
    "    \"POM\": \"POM\",\n",
    "    \"LCC\": \"LCC\",\n",
    "    \"UMKLM\": \"Malaysian Parkinson's Genetics Cohort - UMKLM\",\n",
    "    \"GPARKGR\": \"GenePARKGR\",\n",
    "    \"PROPARK\": \"ProPark\",\n",
    "    \"SP-BCN2\": \"Sp_BCN2 - University Hospital MÃºtua de Terrassa\",\n",
    "    \"SP-SNT\": \"Sp_SNT - University Hospital Marques de Valdecilla-IDIVAL\",\n",
    "    \"BIOFIND\": \"BioFIND\",\n",
    "    \"BLAACPD-KPM\": \"BLAACPD - Kaiser Permanente\",\n",
    "    \"BLAACPD-RUSH\": \"BLAACPD - Rush University\",\n",
    "    \"BLAACPD-UAB\": \"BLAACPD - University of Alabama at Birmingham\",\n",
    "    \"BLAACPD-UC\": \"BLAACPD - University of Chicago\",\n",
    "    \"CALYPSO\": \"CALYPSO\",\n",
    "    \"MDGAP-EBB\": \"MD GAP - Edinburgh Brain Bank - University of Edinburgh\",\n",
    "    \"PFP\": \"PFP\",\n",
    "    \"PPMI\": \"PPMI\",\n",
    "    \"PROBAND\": \"PROBAND\",\n",
    "    \"SYNAPS-KZ\": \"SYNAPS-KZ\",\n",
    "    \"VIPD\": \"VIPD\",\n",
    "    \"CANDAS\": \"CANDAS\",\n",
    "    \"QPP\": \"QPP\",\n",
    "    \"UGUPD\": \"UGCPD\",\n",
    "    \"MDGAP-MBB\": \"MD GAP - Victorian Brain Bank\",\n",
    "    \"GENEFINDER\": \"GeneFinder\",\n",
    "    \"OPDC\": \"OPDC Discovery Cohort\",\n",
    "    \"PROMO\": \"PROMO-USTP\",\n",
    "    \"PROATYP\": \"proATYP\",\n",
    "    \"MDGAP-IBB\": \"MD GAP - Newcastle Brain Bank - University of Newcastle\",\n",
    "    \"SP-GRANADA\": \"ProtectMove - University of Sciences and Technology, Pakistan\",\n",
    "    \"BENHAMIDA\": \"Clinicogenetic Study of PD in Tunisia\",\n",
    "    \"ECMPDC\": None,\n",
    "    \"ELPD\": \"ELPD Project\",\n",
    "    \"PDSGR\": \"PDS\",\n",
    "    \"PREDICTPD\": \"PREDICT-PD\",\n",
    "    \"AUTH\": \"AUTH - GP2 study\",\n",
    "    \"PDSTAT\": \"PD STAT\",\n",
    "    \"CPRC\": None,\n",
    "    \"MPBC\": \"MPBC\",\n",
    "    \"SYDPDRC\": \"Sydney PDRC\",\n",
    "    \"APGS\": \"APGS\",\n",
    "    \"SYDBB\": \"Sydney Brain Bank - SYDBB\",\n",
    "    \"IPDGCEA-JKU\": None,\n",
    "    \"ADEPTPD\": \"ADepT-PD\",\n",
    "    \"PDMDC\": \"PDMDC/NUgene\",\n",
    "    \"GARVAN\": None,\n",
    "    \"KUL\": \"Genotype-Phenotype Correlations in PD and Related Movement Disorders - KUL\",\n",
    "    \"PDMDCB\": None,\n",
    "    \"TUEPAC\": \"Tuepac + TREND\",\n",
    "    \"LUXPARK\": \"LUX-Park\",\n",
    "    \"BIOPARK\": None,\n",
    "    \"IMMLISBON\": None,\n",
    "    \"PDGEN\": \"PDGENEration\",\n",
    "    \"DATATOP\": \"DATATOP\",\n",
    "    \"SZTE\": \"SZTE\",\n",
    "    \"TRD\": None,\n",
    "    \"MDGAP-SW\": \"MD GAP - South West Dementia Brain Bank - Southmead Hospital\",\n",
    "    \"S4\": \"S4\",\n",
    "    \"NETPDLS1\": \"NET-PD\",\n",
    "    \"DENOPA\": \"DeNoPa\",\n",
    "    \"NTUH\": \"IPDGC East Asia - National Taiwan University Hospital\",\n",
    "    \"BRFAA\": \"Biological markers in Parkinson's disease - BRFAA\",\n",
    "    \"BUE\": \"BUE - GP2\",\n",
    "    \"LARGEPD\": \"LARGE PD - FundaciÃ³n Valle del Lili\",\n",
    "    \"BBDP\": \"Arizona Brain Bank Brain and body donation programme (BBDP)\",\n",
    "    \"CINCINNATI\": \"PPMI - University of Cincinnati\",\n",
    "    \"MMCTU\": \"Genotype-Phenotype Correlations in Nepalese PD Population - MMCTU\",\n",
    "    \"PERUGIA\": None,\n",
    "    \"STELLENBOS\": None,\n",
    "    \"CHULALONGK\": \"IPDGC East Asia - Chulalongkorn Centre of Excellence for Parkinson's Disease & Related Disorders\",\n",
    "    \"MDGAP-NWCSTL\": \"MD GAP - Newcastle Brain Bank - University of Newcastle\",\n",
    "    \"KFSH\": \"The Molecular Basis of Movement Disorders in Saudis - KFSH\",\n",
    "    \"PAGE\": \"PAGE\",\n",
    "    \"SYNAPS-GR\": \"SYNAPS-GR\",\n",
    "    \"PDCARE\": \"PDCARE\",\n",
    "    \"RAPSODI\": \"RAPSODI\",\n",
    "    \"YOPDFL\": \"YOPDFL (Young Onset Parkinson's Disease)\",\n",
    "    \"FDSANTIAGO\": None,\n",
    "    \"MULODZ\": None,\n",
    "    \"TONIC-PERRON\": \"TONiC PD\",\n",
    "    \"KOC\": \"GP2 - Koc University\",\n",
    "    \"JGUM\": \"JGUM\",\n",
    "    \"PATRAS\": \"PATRAS\",\n",
    "    \"NUGENE\": \"NU Gene\",\n",
    "    \"CDPDGR\": None,\n",
    "    \"JENPARK\": \"Jena Parkinson Study\",\n",
    "    \"UROMETV\": None,\n",
    "    \"AHEPA\": None,\n",
    "    \"CHCHPD\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"CIDR\": None,\n",
    "    \"HIPDR\": \"HIPDR\",\n",
    "    \"INKOSI\": \"IPDGC Africa - Muhimbili National Hospital\",\n",
    "    \"KOLLING\": \"Kolling PD\",\n",
    "    \"SP-BCN1\": None,\n",
    "    \"SEOULNUH\": \"GP2 - Seoul National University Hospital\",\n",
    "    \"NROSF2U\": None,\n",
    "    \"BLAACPD-LSU\": \"BLAACPD - Louisiana State University\",\n",
    "    \"IMMUNEPD\": \"Innate and Adaptive Immunity in PD\",\n",
    "    \"PROMO-EPIPARK\": \"PROMO-EPIPARK\",\n",
    "    \"PROMO-GENEPARK\": \"PROMO-GENEPARK\",\n",
    "    \"PROMO-ROPAD\": \"PROMO-ROPAD\",\n",
    "    \"PROMO-SFB134\": \"PROMO-SFB134\",\n",
    "    \"PROMO-SYSMED\": \"PROMO-SYSMEDPD\",\n",
    "    \"OXQUIP\": \"Oxford Study of Quantification in Parkinsonism\",\n",
    "    \"PROMO-USTP\": \"PROMO-USTP\",\n",
    "    \"MUVIENNA\": None,\n",
    "    \"CATANZARO\": None,\n",
    "    \"QMAP\": None,\n",
    "    \"CATPD\": \"Central Asian and Transcaucasian PD (CAT-PD)\",\n",
    "    \"IPDGCAF-NG\": \"IPDGC Africa - The Nigerian Parkinson Disease Research Network\",\n",
    "    \"IPDGCEA-JU\": \"IPDGC East Asia - Juntendo University\",\n",
    "    \"TUMBB\": \"Biobank for Diseases of the Peripheral and Central Nervous System - TUMBB\",\n",
    "    \"TUMPD\": \"Genetics of Parkinson's Disease - TUMPD\",\n",
    "    \"ANANEOS\": \"ANANEOS Study\",\n",
    "    \"MDGAP-CAMBRIDGE\": \"MD GAP - Cambridge\",\n",
    "    \"MDGAP-OXF\": \"MD GAP - OXF\",\n",
    "    \"SANGERARDO\": None,\n",
    "    \"VETPD\": \"VET-PD\",\n",
    "    \"BERNGEN\": None,\n",
    "    \"GEMELLI\": None,\n",
    "    \"SINAI\": None,\n",
    "    \"NMDAT\": \"The Norwegian Parkwest Study\",\n",
    "    \"PARKWEST\": \"The Norwegian Parkwest Study\",\n",
    "    \"PROSPECT\": \"PROSPECT\",\n",
    "    \"TARTU\": None,\n",
    "    \"TEHRAN\": None,\n",
    "    \"KIEL\": None,\n",
    "    \"UMUDALL\": \"UMUDALL\",\n",
    "    \"YAGUE\": None,\n",
    "    \"MAYO\": None,\n",
    "    \"TORONTO\": None,\n",
    "    \"AMSBB\": None,\n",
    "    \"MDGAP-VBB\": \"MD GAP - Victorian Brain Bank\",\n",
    "    \"IRCCSN\": None,\n",
    "    \"PDSA\": None,\n",
    "    \"PROBEPD\": \"PROBEPD\",\n",
    "    \"RDIPD\": None,\n",
    "    \"TLVMC\": None,\n",
    "    \"ANKARA\": None,\n",
    "    \"DUPARC\": None,\n",
    "    \"ESTBB\": None,\n",
    "    \"PUJBC\": None,\n",
    "    \"ANDPD\": \"ANDPD\",\n",
    "    \"CMC\": None,\n",
    "    \"SAREMA\": None,\n",
    "    \"MONPAR\": None,\n",
    "    \"NGIPDO\": None,\n",
    "    \"IPDGCEA-MU\": None,\n",
    "    \"IPDGCEA-UKM\": \"IPDGC East Asia - UKM\",\n",
    "    \"DODONA\": None,\n",
    "    \"AARHUS\": None,\n",
    "    \"PROMO-CARRIERS\": \"PROMO-CARRIERS\",\n",
    "    \"CEGEMOD\": None,\n",
    "    \"HAWAIIPD\": \"HAWAII PD\",\n",
    "    \"JUNT\": \"IPDGC East Asia - Juntendo University\",\n",
    "    \"HSPH\": None,\n",
    "    \"YMS\": \"Yale Medical School\",\n",
    "    \"IGECND\": \"IGECND\",\n",
    "    \"NGRC\": None,\n",
    "    \"SILESIA\": None,\n",
    "    \"KRAKOW\": None,\n",
    "    \"BRNO\": None,\n",
    "    \"COLOMBO\": None,\n",
    "    \"DHAKA\": \"GP2 - Bangladesh\",\n",
    "    \"ADRC\": None,\n",
    "    \"SILAD\": None,\n",
    "    \"STANPD\": None,\n",
    "    \"UGCPD\": \"UGCPD\",\n",
    "    \"GENNIS\": None,\n",
    "    \"AAPDGC\": None,\n",
    "    \"AHS\": None,\n",
    "    \"IDIBAPS\": None,\n",
    "    \"FLENI\": None,\n",
    "    \"EPIC\": None,\n",
    "    \"CCFPD\": None,\n",
    "    \"GRACE\": None,\n",
    "    \"NYSCF\": None,\n",
    "    \"PARS\": None,\n",
    "    \"PEG\": None,\n",
    "    \"SAMC\": None,\n",
    "    \"REGARDS\": \"REasons for Geographic and Racial Differences in Stroke (REGARDS) Project\",\n",
    "    \"SAUD\": None,\n",
    "    \"COBB\": None,\n",
    "    \"PDCGC\": None,\n",
    "    \"SANTPAU\": None,\n",
    "    \"TRAPCAF\": None,\n",
    "    \"NZPEGS\": None,\n",
    "    \"NITHAI\": None,\n",
    "    \"RTGRS\": None,\n",
    "    \"PROGENI\": None,\n",
    "    \"BIOPD\": None,\n",
    "    \"BOLOGNA\": None,\n",
    "    \"BPS\": None,\n",
    "    \"CABBS\": None,\n",
    "    \"CLGN\": None,\n",
    "    \"CUHK\": None,\n",
    "    \"GENCI\": None,\n",
    "    \"GPX\": None,\n",
    "    \"HEBA-BRC\": None,\n",
    "    \"HEBA-INNS\": None,\n",
    "    \"HEBA-KASSEL\": None,\n",
    "    \"INECO\": None,\n",
    "    \"INNS\": None,\n",
    "    \"MONT\": None,\n",
    "    \"NORTHSHORE\": None,\n",
    "    \"PARAGEN\": None,\n",
    "    \"PROLBD\": None,\n",
    "    \"RIMU\": None,\n",
    "    \"VAIBB\": \"Van Andel Institute Brain Bank\",\n",
    "    \"WEHI\": None,\n",
    "    \"CANDAS-PDGEN\": None,\n",
    "    \"PARAGEN-MON\": None,\n",
    "    \"UCIPARK\": \"PPMI - University of California San Diego\",\n",
    "    \"UCIPROS\": None,\n",
    "    \"NHGRI\": None,\n",
    "    \"ACCESSPD\": \"ACCESSPD\",\n",
    "    \"CAGL\": None,\n",
    "    \"GPS\": \"Genomes Project\",\n",
    "    \"NIMH\": \"National Institute of Mental Health and Neurosciences\",\n",
    "    \"REMANT\": None,\n",
    "    \"REMEL\": None,\n",
    "    \"UDINE\": None,\n",
    "    \"NGP\": None,\n",
    "    \"REMEDIO\": None,\n",
    "    \"PAIR-PSP\": None,\n",
    "    \"PDBIOMARKERS\": None,\n",
    "    \"NAFPD\": None,\n",
    "    \"FNPREG\": None,\n",
    "    \"GED\": None,\n",
    "    \"PDGEN-CLEM\": None,\n",
    "    \"SCANPSP\": \"SCANPSP - Study of Comprehensive ANd Multimodal Marker-based Cohort of PSP\",\n",
    "    \"MDGAP-DEMENTIA\": None,\n",
    "    \"EXENATIDEPD3\": None,\n",
    "    \"GPDC\": None,\n",
    "    \"NORBANK\": None,\n",
    "    \"NYBB\": None,\n",
    "    \"NYPUM/PARKNY\": None,\n",
    "    \"SAMSUNG\": None,\n",
    "    \"UWBRAIN\": None,\n",
    "    \"SUREPD2\": None,\n",
    "    \"UNIBS\": None,\n",
    "    \"IUC\": None,\n",
    "    \"CANDAS-SMPD\": None,\n",
    "    \"BURGOS\": \"Local Study - Burgos\",\n",
    "    \"PPRBD\": None,\n",
    "    \"EMORY\": None,\n",
    "    \"CANPARK\": None,\n",
    "    \"TAS\": None,\n",
    "    \"ISLAND\": None,\n",
    "    \"CRUCES\": None,\n",
    "    \"NRIBBB\": None,\n",
    "    \"NBB-HVD\": None,\n",
    "    \"NBB-SINAI\": None,\n",
    "    \"NBB-SPLVD\": None,\n",
    "    \"NBB-MLND\": None,\n",
    "    \"NBB-MIAMI\": None,\n",
    "    \"NBB-PBG\": None,\n",
    "    \"BELGRADE\": None,\n",
    "    \"MHHBB\": None,\n",
    "    \"SLOWSPEED\": None,\n",
    "    \"FMGC\": None,\n",
    "    \"IRAMS\": None,\n",
    "    \"GENOT\": None,\n",
    "    \"UTSW\": None,\n",
    "    \"ADPAR\": None,\n",
    "    \"FAIRPARK\": None,\n",
    "    \"GBA1NL\": None,\n",
    "    \"PDNEF\": None,\n",
    "    \"SP-DON\": None,\n",
    "\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "df_sub = df.loc[:,[\n",
    "    \"Study Code\",\n",
    "    \"Monogenic / Complex / Mixed\",\n",
    "    \"City\",\n",
    "    \"Geographic locality [free text]\",\n",
    "    \"DNA samples attempted [N]\",\n",
    "    \"Total samples passing basic QC [N]\",\n",
    "    \"Failed call rates [N < 95%]\",\n",
    "    \"Failed sex check [N with contradictory reported versus genetic sex]\",\n",
    "    \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\",\n",
    "    \"Duplicate samples [N, IBD estiamtes > 95%]\",\n",
    "    \"African (AFR)\",\n",
    "    \"African admixed (AAC)\",\n",
    "    \"Ashkenzai (AJ)\",\n",
    "    \"East Asian (EAS)\",\n",
    "    \"European (EUR)\",\n",
    "    \"Finnish (FIN)\",\n",
    "    \"LatinX (AMR)\",\n",
    "    \"South Asian (SAS)\",\n",
    "    \"Central Asian (CAS)\",\n",
    "    \"Middle Eastern (MDE)\",\n",
    "    \"Complex Admixture (CAH)\",\n",
    "    \"Total\",\n",
    "    \"Imputation panels(s) [Free text]\",\n",
    "    \"Imputation complete for current samples [0 = no, 1 = yes]\",\n",
    "    \"Full data [0 = only summary stats, 1 = genotypes sharable]\",\n",
    "    \"GDPR [0 = no European restriction, 1 = limited by GDPR]\",\n",
    "    \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\"\n",
    "]]\n",
    "\n",
    "rename_dict = {\n",
    "    \"Study Code\": \"study_code\",\n",
    "    \"Monogenic / Complex / Mixed\": \"monogenic_complex_mixed\",\n",
    "    \"City\": \"city\",\n",
    "    \"Geographic locality [free text]\": \"geographic_locality\",\n",
    "    \"DNA samples attempted [N]\": \"n_dna_samples_attempted\",\n",
    "    \"Total samples passing basic QC [N]\": \"total_qc_pass\",\n",
    "    \"Failed call rates [N < 95%]\": \"callrate_fails\",\n",
    "    \"Failed sex check [N with contradictory reported versus genetic sex]\": \"sex_fails\",\n",
    "    \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\": \"het_fails\",\n",
    "    \"Duplicate samples [N, IBD estiamtes > 95%]\": \"duplicates\",\n",
    "    \"African (AFR)\": \"AFR\",\n",
    "    \"African admixed (AAC)\": \"AAC\",\n",
    "    \"Ashkenzai (AJ)\": \"AJ\",\n",
    "    \"East Asian (EAS)\": \"EAS\",\n",
    "    \"European (EUR)\": \"EUR\",\n",
    "    \"Finnish (FIN)\": \"FIN\",\n",
    "    \"LatinX (AMR)\": \"AMR\",\n",
    "    \"South Asian (SAS)\": \"SAS\",\n",
    "    \"Central Asian (CAS)\": \"CAS\",\n",
    "    \"Middle Eastern (MDE)\": \"MDE\",\n",
    "    \"Complex Admixture (CAH)\": \"CAH\",\n",
    "    \"Total\": \"total\",\n",
    "    \"Imputation panels(s) [Free text]\": \"imputation_panel\",\n",
    "    \"Imputation complete for current samples [0 = no, 1 = yes]\": \"imputation_complete\",\n",
    "    \"Full data [0 = only summary stats, 1 = genotypes sharable]\": \"genotypes_shareable\",\n",
    "    \"GDPR [0 = no European restriction, 1 = limited by GDPR]\": \"gdpr\",\n",
    "    \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\": \"site\"\n",
    "}\n",
    "\n",
    "df_sub.rename(columns=rename_dict, inplace=True)\n",
    "\n",
    "df_out = df_sub.fillna({\n",
    "    'study_code': '',\n",
    "    'monogenic_complex_mixed': '',\n",
    "    'city': '',\n",
    "    'geographic_locality': '',\n",
    "    'n_dna_samples_attempted': 0,\n",
    "    'total_qc_pass': 0,\n",
    "    'callrate_fails': 0,\n",
    "    'sex_fails': 0,\n",
    "    'het_fails': 0,\n",
    "    'duplicates': 0,\n",
    "    'AFR': 0,\n",
    "    'AAC': 0,\n",
    "    'AJ': 0,\n",
    "    'EAS': 0,\n",
    "    'EUR': 0,\n",
    "    'FIN': 0,\n",
    "    'AMR': 0,\n",
    "    'SAS': 0,\n",
    "    'CAS': 0,\n",
    "    'MDE': 0,\n",
    "    'CAH': 0,\n",
    "    'total': 0,\n",
    "    'genotypes_shareable': 0,\n",
    "    'imputation_complete': 0,\n",
    "    'imputation_panel': '',\n",
    "    'gdpr': 0,\n",
    "    'site': ''\n",
    "})\n",
    "\n",
    "numeric_columns = [\n",
    "    'n_dna_samples_attempted', 'total_qc_pass', 'callrate_fails', 'sex_fails', \n",
    "    'het_fails', 'duplicates', 'AFR', 'AAC', 'AJ', 'EAS', 'EUR', 'FIN', \n",
    "    'AMR', 'SAS', 'CAS', 'MDE', 'CAH', 'total', 'genotypes_shareable', \n",
    "    'imputation_complete', 'gdpr'\n",
    "]\n",
    "\n",
    "df_out[numeric_columns] = df_out[numeric_columns].astype(int)\n",
    "df_out.to_csv('data/genotracker_clean.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_code</th>\n",
       "      <th>monogenic_complex_mixed</th>\n",
       "      <th>city</th>\n",
       "      <th>geographic_locality</th>\n",
       "      <th>n_dna_samples_attempted</th>\n",
       "      <th>total_qc_pass</th>\n",
       "      <th>callrate_fails</th>\n",
       "      <th>sex_fails</th>\n",
       "      <th>het_fails</th>\n",
       "      <th>duplicates</th>\n",
       "      <th>...</th>\n",
       "      <th>CAS Case</th>\n",
       "      <th>CAS Control</th>\n",
       "      <th>CAS Other</th>\n",
       "      <th>MDE Case</th>\n",
       "      <th>MDE Control</th>\n",
       "      <th>MDE Other</th>\n",
       "      <th>CAH Case</th>\n",
       "      <th>CAH Control</th>\n",
       "      <th>CAH Other</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AUTH</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Thessaloniki</td>\n",
       "      <td>Greece</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CATPD</td>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>Asia/Oceania</td>\n",
       "      <td>665</td>\n",
       "      <td>633</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>268</td>\n",
       "      <td>50</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>183</td>\n",
       "      <td>2</td>\n",
       "      <td>22</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>1268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DODONA</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Chicago</td>\n",
       "      <td>United States</td>\n",
       "      <td>747</td>\n",
       "      <td>722</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGUM</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Munich / Mainz</td>\n",
       "      <td>Germany</td>\n",
       "      <td>564</td>\n",
       "      <td>559</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Edinburgh, London</td>\n",
       "      <td>UK</td>\n",
       "      <td>212</td>\n",
       "      <td>176</td>\n",
       "      <td>28</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MDGAP-IBB</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>London</td>\n",
       "      <td>UK</td>\n",
       "      <td>386</td>\n",
       "      <td>350</td>\n",
       "      <td>12</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MDGAP-OXF</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Oxford</td>\n",
       "      <td>UK</td>\n",
       "      <td>41</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MDGAP-QSBB</td>\n",
       "      <td>Complex</td>\n",
       "      <td>London</td>\n",
       "      <td>UK</td>\n",
       "      <td>1194</td>\n",
       "      <td>1021</td>\n",
       "      <td>55</td>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MDGAP-SW</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Bristol</td>\n",
       "      <td>England</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NZPEGS</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Christchurch</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>345</td>\n",
       "      <td>329</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PDGNRTN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Miami, FL</td>\n",
       "      <td>USA</td>\n",
       "      <td>4694</td>\n",
       "      <td>3791</td>\n",
       "      <td>239</td>\n",
       "      <td>352</td>\n",
       "      <td>0</td>\n",
       "      <td>296</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PERUGIA</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Perugia</td>\n",
       "      <td>Italy</td>\n",
       "      <td>100</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PROBAND</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Glasgow</td>\n",
       "      <td>UK</td>\n",
       "      <td>41</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PROSPECT</td>\n",
       "      <td>Complex</td>\n",
       "      <td>London</td>\n",
       "      <td>England</td>\n",
       "      <td>299</td>\n",
       "      <td>212</td>\n",
       "      <td>16</td>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>RAPSODI</td>\n",
       "      <td>Complex</td>\n",
       "      <td>London</td>\n",
       "      <td>UK</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>REGARDS</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>United States</td>\n",
       "      <td>5587</td>\n",
       "      <td>5370</td>\n",
       "      <td>113</td>\n",
       "      <td>57</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1746</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16 rows Ã— 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    study_code monogenic_complex_mixed               city geographic_locality  \\\n",
       "0         AUTH                 Complex       Thessaloniki              Greece   \n",
       "1        CATPD                 Complex                  0        Asia/Oceania   \n",
       "2       DODONA                   Mixed            Chicago       United States   \n",
       "3         JGUM                 Complex     Munich / Mainz             Germany   \n",
       "4    MDGAP-EBB                 Complex  Edinburgh, London                  UK   \n",
       "5    MDGAP-IBB                   Mixed             London                  UK   \n",
       "6    MDGAP-OXF                 Complex             Oxford                  UK   \n",
       "7   MDGAP-QSBB                 Complex             London                  UK   \n",
       "8     MDGAP-SW                 Complex            Bristol             England   \n",
       "9       NZPEGS                   Mixed       Christchurch         New Zealand   \n",
       "10     PDGNRTN                   Mixed         Miami, FL                  USA   \n",
       "11     PERUGIA                   Mixed            Perugia               Italy   \n",
       "12     PROBAND                   Mixed            Glasgow                  UK   \n",
       "13    PROSPECT                 Complex             London             England   \n",
       "14     RAPSODI                 Complex             London                  UK   \n",
       "15     REGARDS                 Complex         Birmingham       United States   \n",
       "\n",
       "    n_dna_samples_attempted  total_qc_pass  callrate_fails  sex_fails  \\\n",
       "0                        88             88               0          0   \n",
       "1                       665            633              14         13   \n",
       "2                       747            722              16          1   \n",
       "3                       564            559               0          4   \n",
       "4                       212            176              28          6   \n",
       "5                       386            350              12         21   \n",
       "6                        41             10               0         31   \n",
       "7                      1194           1021              55         85   \n",
       "8                         0              0               0          0   \n",
       "9                       345            329              15          1   \n",
       "10                     4694           3791             239        352   \n",
       "11                      100             93               0          7   \n",
       "12                       41             39               1          0   \n",
       "13                      299            212              16         66   \n",
       "14                        0              0               0          0   \n",
       "15                     5587           5370             113         57   \n",
       "\n",
       "    het_fails  duplicates  ...  CAS Case  CAS Control  CAS Other  MDE Case  \\\n",
       "0           0           0  ...         0            0          0         4   \n",
       "1           0           5  ...       268           50          2        12   \n",
       "2           0           7  ...         1            0          0         8   \n",
       "3           0           1  ...         1            1          0        13   \n",
       "4           0           2  ...         0            0          0         0   \n",
       "5           0           2  ...         0            0          0         0   \n",
       "6           0           0  ...         0            0          1         0   \n",
       "7           0          16  ...         0            0          0         0   \n",
       "8           0           0  ...         0            0          0         0   \n",
       "9           0           0  ...         0            0          0         0   \n",
       "10          0         296  ...         6            0          0         4   \n",
       "11          0           0  ...         0            0          0         0   \n",
       "12          0           1  ...         0            0          0         0   \n",
       "13          0           5  ...         0            0          0         0   \n",
       "14          0           0  ...         0            0          0         1   \n",
       "15          0           5  ...         0            0          0         0   \n",
       "\n",
       "    MDE Control  MDE Other  CAH Case  CAH Control  CAH Other  Total  \n",
       "0             0          0         0            0          0    176  \n",
       "1           183          2        22           24          0   1268  \n",
       "2             0          0         2            0          0   1456  \n",
       "3             1          0         4            1          0   1120  \n",
       "4             0          0         0            0          0     62  \n",
       "5             0          0         0            0          0    926  \n",
       "6             0          0         0            0          0    350  \n",
       "7             0          0         0            0          0    608  \n",
       "8             0          0         0            0          0    232  \n",
       "9             0          0         0            0          0    658  \n",
       "10            0          0        29            0          0   2972  \n",
       "11            0          0         0            0          0    186  \n",
       "12            0          0         0            0          0     80  \n",
       "13            0          0         0            0          1    334  \n",
       "14            0          0         0            0          0    188  \n",
       "15            1          0         0            2          0   1746  \n",
       "\n",
       "[16 rows x 61 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ancestry_df = pd.read_csv('data/FEB_2024_cohort_ancestry_phenotype.csv')\n",
    "ancestry_df.rename(columns={'Cohort':'study_code'}, inplace=True)\n",
    "df_out.merge(ancestry_df, on='study_code', how='right')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_code</th>\n",
       "      <th>monogenic_complex_mixed</th>\n",
       "      <th>city</th>\n",
       "      <th>geographic_locality</th>\n",
       "      <th>n_dna_samples_attempted</th>\n",
       "      <th>total_qc_pass</th>\n",
       "      <th>callrate_fails</th>\n",
       "      <th>sex_fails</th>\n",
       "      <th>het_fails</th>\n",
       "      <th>duplicates</th>\n",
       "      <th>...</th>\n",
       "      <th>SAS</th>\n",
       "      <th>CAS</th>\n",
       "      <th>MDE</th>\n",
       "      <th>CAH</th>\n",
       "      <th>total</th>\n",
       "      <th>imputation_panel</th>\n",
       "      <th>imputation_complete</th>\n",
       "      <th>genotypes_shareable</th>\n",
       "      <th>gdpr</th>\n",
       "      <th>site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BCM</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Houston,TX</td>\n",
       "      <td>USA</td>\n",
       "      <td>809</td>\n",
       "      <td>788</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>788</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CORIELL</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Camden, NJ</td>\n",
       "      <td>USA</td>\n",
       "      <td>8982</td>\n",
       "      <td>8718</td>\n",
       "      <td>140</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>40</td>\n",
       "      <td>...</td>\n",
       "      <td>40</td>\n",
       "      <td>8</td>\n",
       "      <td>48</td>\n",
       "      <td>167</td>\n",
       "      <td>8718</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGNRTN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Miami, FL</td>\n",
       "      <td>USA</td>\n",
       "      <td>4694</td>\n",
       "      <td>3791</td>\n",
       "      <td>239</td>\n",
       "      <td>352</td>\n",
       "      <td>0</td>\n",
       "      <td>296</td>\n",
       "      <td>...</td>\n",
       "      <td>45</td>\n",
       "      <td>16</td>\n",
       "      <td>21</td>\n",
       "      <td>174</td>\n",
       "      <td>3791</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Fulgent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UMD</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Baltimore, MD</td>\n",
       "      <td>USA</td>\n",
       "      <td>479</td>\n",
       "      <td>441</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>441</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PROSPOS</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Oslo</td>\n",
       "      <td>Norway</td>\n",
       "      <td>96</td>\n",
       "      <td>95</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Fulgent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>FMGC</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>IRAMS</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>GENOT</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Tunis</td>\n",
       "      <td>Tunisia</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>UTSW</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Dallas</td>\n",
       "      <td>United States</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>ADPAR</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bucharest</td>\n",
       "      <td>Romania</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    study_code monogenic_complex_mixed           city geographic_locality  \\\n",
       "0          BCM                 Complex     Houston,TX                 USA   \n",
       "1      CORIELL                 Complex     Camden, NJ                 USA   \n",
       "2      PDGNRTN                   Mixed     Miami, FL                  USA   \n",
       "3          UMD                 Complex  Baltimore, MD                 USA   \n",
       "4      PROSPOS                 Complex           Oslo              Norway   \n",
       "..         ...                     ...            ...                 ...   \n",
       "306       FMGC                   Mixed       Bordeaux              France   \n",
       "307      IRAMS                   Mixed       Bordeaux              France   \n",
       "308      GENOT                   Mixed          Tunis             Tunisia   \n",
       "309       UTSW                   Mixed         Dallas       United States   \n",
       "310      ADPAR                   Mixed      Bucharest             Romania   \n",
       "\n",
       "     n_dna_samples_attempted  total_qc_pass  callrate_fails  sex_fails  \\\n",
       "0                        809            788               7         12   \n",
       "1                       8982           8718             140         56   \n",
       "2                       4694           3791             239        352   \n",
       "3                        479            441               6         15   \n",
       "4                         96             95               0          1   \n",
       "..                       ...            ...             ...        ...   \n",
       "306                        0              0               0          0   \n",
       "307                        0              0               0          0   \n",
       "308                        0              0               0          0   \n",
       "309                        0              0               0          0   \n",
       "310                        0              0               0          0   \n",
       "\n",
       "     het_fails  duplicates  ...  SAS  CAS  MDE  CAH  total  imputation_panel  \\\n",
       "0            0           2  ...   13    5    4    5    788            TOPMED   \n",
       "1            1          40  ...   40    8   48  167   8718            TOPMED   \n",
       "2            0         296  ...   45   16   21  174   3791            TOPMED   \n",
       "3            0          17  ...    5    2    2    2    441            TOPMED   \n",
       "4            0           0  ...    0    0    0    0     95            TOPMED   \n",
       "..         ...         ...  ...  ...  ...  ...  ...    ...               ...   \n",
       "306          0           0  ...    0    0    0    0      0                 0   \n",
       "307          0           0  ...    0    0    0    0      0                 0   \n",
       "308          0           0  ...    0    0    0    0      0                 0   \n",
       "309          0           0  ...    0    0    0    0      0                 0   \n",
       "310          0           0  ...    0    0    0    0      0                 0   \n",
       "\n",
       "     imputation_complete  genotypes_shareable  gdpr     site  \n",
       "0                      1                    1     0      NIH  \n",
       "1                      1                    1     0      NIH  \n",
       "2                      1                    1     0  Fulgent  \n",
       "3                      1                    1     0      NIH  \n",
       "4                      1                    1     1  Fulgent  \n",
       "..                   ...                  ...   ...      ...  \n",
       "306                    0                    0     0        0  \n",
       "307                    0                    0     0        0  \n",
       "308                    0                    0     0        0  \n",
       "309                    0                    0     0        0  \n",
       "310                    0                    0     0        0  \n",
       "\n",
       "[311 rows x 27 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   id study_code monogenic_complex_mixed           city geographic_locality  \\\n",
      "0   1        BCM                 Complex     Houston,TX                 USA   \n",
      "1   2    CORIELL                 Complex     Camden, NJ                 USA   \n",
      "2   3    PDGNRTN                   Mixed     Miami, FL                  USA   \n",
      "3   4        UMD                 Complex  Baltimore, MD                 USA   \n",
      "4   5    PROSPOS                 Complex           Oslo              Norway   \n",
      "\n",
      "   n_dna_samples_attempted  total_qc_pass  callrate_fails  sex_fails  \\\n",
      "0                      809            788               7         12   \n",
      "1                     8982           8718             140         56   \n",
      "2                     4694           3791             239        352   \n",
      "3                      479            441               6         15   \n",
      "4                       96             95               0          1   \n",
      "\n",
      "   het_fails  ...  CAS  MDE  CAH  total  genotyping_complete  \\\n",
      "0          0  ...    5    4    5    788                    0   \n",
      "1          1  ...    8   48  167   8718                    0   \n",
      "2          0  ...   16   21  174   3791                    0   \n",
      "3          0  ...    2    2    2    441                    0   \n",
      "4          0  ...    0    0    0     95                    0   \n",
      "\n",
      "   imputation_panel  imputation_complete  genotypes_shareable  gdpr     site  \n",
      "0            TOPMED                    1                    1     0      NIH  \n",
      "1            TOPMED                    1                    1     0      NIH  \n",
      "2            TOPMED                    1                    1     0  Fulgent  \n",
      "3            TOPMED                    1                    1     0      NIH  \n",
      "4            TOPMED                    1                    1     1  Fulgent  \n",
      "\n",
      "[5 rows x 29 columns]\n",
      "               id  n_dna_samples_attempted  total_qc_pass  callrate_fails  \\\n",
      "count  311.000000               311.000000     311.000000      311.000000   \n",
      "mean   156.000000               189.302251     175.440514        5.173633   \n",
      "std     89.922189               761.131042     716.178651       24.136081   \n",
      "min      1.000000                 0.000000       0.000000        0.000000   \n",
      "25%     78.500000                 0.000000       0.000000        0.000000   \n",
      "50%    156.000000                 0.000000       0.000000        0.000000   \n",
      "75%    233.500000                19.000000      18.500000        0.000000   \n",
      "max    311.000000              8982.000000    8718.000000      239.000000   \n",
      "\n",
      "        sex_fails   het_fails  duplicates          AFR         AAC  \\\n",
      "count  311.000000  311.000000  311.000000   311.000000  311.000000   \n",
      "mean     4.427653    0.009646    3.559486     8.501608    3.578778   \n",
      "std     22.928650    0.097898   20.679010    91.034339   41.613731   \n",
      "min      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "25%      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "50%      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "75%      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "max    352.000000    1.000000  296.000000  1403.000000  723.000000   \n",
      "\n",
      "               AJ  ...         AMR         SAS         CAS         MDE  \\\n",
      "count  311.000000  ...  311.000000  311.000000  311.000000  311.000000   \n",
      "mean     8.562701  ...    2.099678    2.073955    2.906752    1.871383   \n",
      "std     62.169345  ...   15.717251   12.055423   28.430918   12.646159   \n",
      "min      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "25%      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "50%      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "75%      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "max    780.000000  ...  195.000000  142.000000  366.000000  197.000000   \n",
      "\n",
      "              CAH        total  genotyping_complete  imputation_complete  \\\n",
      "count  311.000000   311.000000                311.0           311.000000   \n",
      "mean     2.742765   175.440514                  0.0             0.257235   \n",
      "std     22.141293   716.178651                  0.0             0.437814   \n",
      "min      0.000000     0.000000                  0.0             0.000000   \n",
      "25%      0.000000     0.000000                  0.0             0.000000   \n",
      "50%      0.000000     0.000000                  0.0             0.000000   \n",
      "75%      0.000000    18.500000                  0.0             1.000000   \n",
      "max    299.000000  8718.000000                  0.0             1.000000   \n",
      "\n",
      "       genotypes_shareable        gdpr  \n",
      "count           311.000000  311.000000  \n",
      "mean              0.334405    0.202572  \n",
      "std               0.472542    0.402564  \n",
      "min               0.000000    0.000000  \n",
      "25%               0.000000    0.000000  \n",
      "50%               0.000000    0.000000  \n",
      "75%               1.000000    0.000000  \n",
      "max               1.000000    1.000000  \n",
      "\n",
      "[8 rows x 23 columns]\n",
      "id                          int64\n",
      "study_code                 object\n",
      "monogenic_complex_mixed    object\n",
      "city                       object\n",
      "geographic_locality        object\n",
      "n_dna_samples_attempted     int64\n",
      "total_qc_pass               int64\n",
      "callrate_fails              int64\n",
      "sex_fails                   int64\n",
      "het_fails                   int64\n",
      "duplicates                  int64\n",
      "AFR                         int64\n",
      "AAC                         int64\n",
      "AJ                          int64\n",
      "EAS                         int64\n",
      "EUR                         int64\n",
      "FIN                         int64\n",
      "AMR                         int64\n",
      "SAS                         int64\n",
      "CAS                         int64\n",
      "MDE                         int64\n",
      "CAH                         int64\n",
      "total                       int64\n",
      "genotyping_complete         int64\n",
      "imputation_panel           object\n",
      "imputation_complete         int64\n",
      "genotypes_shareable         int64\n",
      "gdpr                        int64\n",
      "site                       object\n",
      "dtype: object\n",
      "Empty DataFrame\n",
      "Columns: [id, study_code, monogenic_complex_mixed, city, geographic_locality, n_dna_samples_attempted, total_qc_pass, callrate_fails, sex_fails, het_fails, duplicates, AFR, AAC, AJ, EAS, EUR, FIN, AMR, SAS, CAS, MDE, CAH, total, genotyping_complete, imputation_panel, imputation_complete, genotypes_shareable, gdpr, site]\n",
      "Index: []\n",
      "\n",
      "[0 rows x 29 columns]\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Create an engine instance\n",
    "engine = create_engine('sqlite:///test.db')\n",
    "\n",
    "# Query to select all data from the genetic_data table\n",
    "query = \"SELECT * FROM genetic_data\"\n",
    "\n",
    "# Execute the query and load the data into a pandas DataFrame\n",
    "df = pd.read_sql(query, engine)\n",
    "\n",
    "# Display the first few rows of the DataFrame\n",
    "print(df.head())\n",
    "\n",
    "# Display summary statistics\n",
    "print(df.describe())\n",
    "\n",
    "# Display the data types of the columns\n",
    "print(df.dtypes)\n",
    "\n",
    "# Filter the data\n",
    "df_filtered = df[df['city'] == 'Boulder']\n",
    "print(df_filtered.head())\n",
    "\n",
    "# # Plot the data\n",
    "# import matplotlib.pyplot as plt\n",
    "\n",
    "# df['n_dna_samples_attempted'].hist()\n",
    "# plt.xlabel('Number of DNA Samples Attempted')\n",
    "# plt.ylabel('Frequency')\n",
    "# plt.title('Distribution of DNA Samples Attempted')\n",
    "# plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>study_code</th>\n",
       "      <th>monogenic_complex_mixed</th>\n",
       "      <th>city</th>\n",
       "      <th>geographic_locality</th>\n",
       "      <th>n_dna_samples_attempted</th>\n",
       "      <th>total_qc_pass</th>\n",
       "      <th>callrate_fails</th>\n",
       "      <th>sex_fails</th>\n",
       "      <th>het_fails</th>\n",
       "      <th>...</th>\n",
       "      <th>CAS</th>\n",
       "      <th>MDE</th>\n",
       "      <th>CAH</th>\n",
       "      <th>total</th>\n",
       "      <th>genotyping_complete</th>\n",
       "      <th>imputation_panel</th>\n",
       "      <th>imputation_complete</th>\n",
       "      <th>genotypes_shareable</th>\n",
       "      <th>gdpr</th>\n",
       "      <th>site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>BCM</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Houston,TX</td>\n",
       "      <td>USA</td>\n",
       "      <td>809</td>\n",
       "      <td>788</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>788</td>\n",
       "      <td>0</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Camden, NJ</td>\n",
       "      <td>USA</td>\n",
       "      <td>8982</td>\n",
       "      <td>8718</td>\n",
       "      <td>140</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>48</td>\n",
       "      <td>167</td>\n",
       "      <td>8718</td>\n",
       "      <td>0</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>PDGNRTN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Miami, FL</td>\n",
       "      <td>USA</td>\n",
       "      <td>4694</td>\n",
       "      <td>3791</td>\n",
       "      <td>239</td>\n",
       "      <td>352</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>21</td>\n",
       "      <td>174</td>\n",
       "      <td>3791</td>\n",
       "      <td>0</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Fulgent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>UMD</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Baltimore, MD</td>\n",
       "      <td>USA</td>\n",
       "      <td>479</td>\n",
       "      <td>441</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>441</td>\n",
       "      <td>0</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>PROSPOS</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Oslo</td>\n",
       "      <td>Norway</td>\n",
       "      <td>96</td>\n",
       "      <td>95</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>0</td>\n",
       "      <td>TOPMED</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Fulgent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>307</td>\n",
       "      <td>FMGC</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>308</td>\n",
       "      <td>IRAMS</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>309</td>\n",
       "      <td>GENOT</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Tunis</td>\n",
       "      <td>Tunisia</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>310</td>\n",
       "      <td>UTSW</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Dallas</td>\n",
       "      <td>United States</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>311</td>\n",
       "      <td>ADPAR</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bucharest</td>\n",
       "      <td>Romania</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id study_code monogenic_complex_mixed           city  \\\n",
       "0      1        BCM                 Complex     Houston,TX   \n",
       "1      2    CORIELL                 Complex     Camden, NJ   \n",
       "2      3    PDGNRTN                   Mixed     Miami, FL    \n",
       "3      4        UMD                 Complex  Baltimore, MD   \n",
       "4      5    PROSPOS                 Complex           Oslo   \n",
       "..   ...        ...                     ...            ...   \n",
       "306  307       FMGC                   Mixed       Bordeaux   \n",
       "307  308      IRAMS                   Mixed       Bordeaux   \n",
       "308  309      GENOT                   Mixed          Tunis   \n",
       "309  310       UTSW                   Mixed         Dallas   \n",
       "310  311      ADPAR                   Mixed      Bucharest   \n",
       "\n",
       "    geographic_locality  n_dna_samples_attempted  total_qc_pass  \\\n",
       "0                   USA                      809            788   \n",
       "1                   USA                     8982           8718   \n",
       "2                   USA                     4694           3791   \n",
       "3                   USA                      479            441   \n",
       "4                Norway                       96             95   \n",
       "..                  ...                      ...            ...   \n",
       "306              France                        0              0   \n",
       "307              France                        0              0   \n",
       "308             Tunisia                        0              0   \n",
       "309       United States                        0              0   \n",
       "310             Romania                        0              0   \n",
       "\n",
       "     callrate_fails  sex_fails  het_fails  ...  CAS  MDE  CAH  total  \\\n",
       "0                 7         12          0  ...    5    4    5    788   \n",
       "1               140         56          1  ...    8   48  167   8718   \n",
       "2               239        352          0  ...   16   21  174   3791   \n",
       "3                 6         15          0  ...    2    2    2    441   \n",
       "4                 0          1          0  ...    0    0    0     95   \n",
       "..              ...        ...        ...  ...  ...  ...  ...    ...   \n",
       "306               0          0          0  ...    0    0    0      0   \n",
       "307               0          0          0  ...    0    0    0      0   \n",
       "308               0          0          0  ...    0    0    0      0   \n",
       "309               0          0          0  ...    0    0    0      0   \n",
       "310               0          0          0  ...    0    0    0      0   \n",
       "\n",
       "     genotyping_complete  imputation_panel  imputation_complete  \\\n",
       "0                      0            TOPMED                    1   \n",
       "1                      0            TOPMED                    1   \n",
       "2                      0            TOPMED                    1   \n",
       "3                      0            TOPMED                    1   \n",
       "4                      0            TOPMED                    1   \n",
       "..                   ...               ...                  ...   \n",
       "306                    0                 0                    0   \n",
       "307                    0                 0                    0   \n",
       "308                    0                 0                    0   \n",
       "309                    0                 0                    0   \n",
       "310                    0                 0                    0   \n",
       "\n",
       "     genotypes_shareable  gdpr     site  \n",
       "0                      1     0      NIH  \n",
       "1                      1     0      NIH  \n",
       "2                      1     0  Fulgent  \n",
       "3                      1     0      NIH  \n",
       "4                      1     1  Fulgent  \n",
       "..                   ...   ...      ...  \n",
       "306                    0     0        0  \n",
       "307                    0     0        0  \n",
       "308                    0     0        0  \n",
       "309                    0     0        0  \n",
       "310                    0     0        0  \n",
       "\n",
       "[311 rows x 29 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "study_code\n",
       "0             19\n",
       "BCM            1\n",
       "EPIC           1\n",
       "IDIBAPS        1\n",
       "AHS            1\n",
       "              ..\n",
       "CHULALONGK     1\n",
       "STELLENBOS     1\n",
       "PERUGIA        1\n",
       "MMCTU          1\n",
       "ADPAR          1\n",
       "Name: count, Length: 293, dtype: int64"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out.study_code.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_code</th>\n",
       "      <th>monogenic_complex_mixed</th>\n",
       "      <th>city</th>\n",
       "      <th>geographic_locality</th>\n",
       "      <th>n_dna_samples_attempted</th>\n",
       "      <th>total_qc_pass</th>\n",
       "      <th>callrate_fails</th>\n",
       "      <th>sex_fails</th>\n",
       "      <th>het_fails</th>\n",
       "      <th>duplicates</th>\n",
       "      <th>...</th>\n",
       "      <th>SAS</th>\n",
       "      <th>CAS</th>\n",
       "      <th>MDE</th>\n",
       "      <th>CAH</th>\n",
       "      <th>total</th>\n",
       "      <th>imputation_panel</th>\n",
       "      <th>imputation_complete</th>\n",
       "      <th>genotypes_shareable</th>\n",
       "      <th>gdpr</th>\n",
       "      <th>site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>0</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Edinburgh</td>\n",
       "      <td>UK</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Sulawesi</td>\n",
       "      <td>Indonesia</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>DZNE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Ankara</td>\n",
       "      <td>Turkey</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Moscow</td>\n",
       "      <td>Russia</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>0</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>London</td>\n",
       "      <td>UK</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Lubeck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>0</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Newcastle</td>\n",
       "      <td>UK</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>0</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Newcastle</td>\n",
       "      <td>UK</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Wilrijk</td>\n",
       "      <td>Belgium</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>DZNE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Tel-Aviv</td>\n",
       "      <td>Israel</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Strasbourg</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Auckland</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Cebu</td>\n",
       "      <td>Philippines</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH, Lubeck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Jakarta</td>\n",
       "      <td>Indonesia</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Lubeck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Vienna</td>\n",
       "      <td>Austria</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>0</td>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>DZNE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Parma</td>\n",
       "      <td>Italy</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Milan</td>\n",
       "      <td>Italy</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Lubeck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>0</td>\n",
       "      <td>Monogenic</td>\n",
       "      <td>Messin</td>\n",
       "      <td>Italy</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Bishkek</td>\n",
       "      <td>Kyrgyzstan</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    study_code monogenic_complex_mixed        city geographic_locality  \\\n",
       "26           0                 Complex   Edinburgh                  UK   \n",
       "98           0               Monogenic    Sulawesi           Indonesia   \n",
       "99           0               Monogenic      Ankara              Turkey   \n",
       "100          0               Monogenic      Moscow              Russia   \n",
       "112          0                   Mixed      London                  UK   \n",
       "115          0                 Complex   Newcastle                  UK   \n",
       "116          0                 Complex   Newcastle                  UK   \n",
       "156          0               Monogenic     Wilrijk             Belgium   \n",
       "158          0               Monogenic    Tel-Aviv              Israel   \n",
       "162          0               Monogenic  Strasbourg              France   \n",
       "165          0               Monogenic    Auckland         New Zealand   \n",
       "180          0               Monogenic        Cebu         Philippines   \n",
       "181          0               Monogenic     Jakarta           Indonesia   \n",
       "185          0               Monogenic      Vienna             Austria   \n",
       "190          0                 Complex           0                   0   \n",
       "210          0               Monogenic       Parma               Italy   \n",
       "233          0               Monogenic       Milan               Italy   \n",
       "269          0               Monogenic      Messin               Italy   \n",
       "276          0                       0     Bishkek          Kyrgyzstan   \n",
       "\n",
       "     n_dna_samples_attempted  total_qc_pass  callrate_fails  sex_fails  \\\n",
       "26                         0              0               0          0   \n",
       "98                         0              0               0          0   \n",
       "99                         0              0               0          0   \n",
       "100                        0              0               0          0   \n",
       "112                        0              0               0          0   \n",
       "115                        0              0               0          0   \n",
       "116                        0              0               0          0   \n",
       "156                        0              0               0          0   \n",
       "158                        0              0               0          0   \n",
       "162                        0              0               0          0   \n",
       "165                        0              0               0          0   \n",
       "180                        0              0               0          0   \n",
       "181                        0              0               0          0   \n",
       "185                        0              0               0          0   \n",
       "190                        0             22               0          0   \n",
       "210                        0              0               0          0   \n",
       "233                        0              0               0          0   \n",
       "269                        0              0               0          0   \n",
       "276                        0              0               0          0   \n",
       "\n",
       "     het_fails  duplicates  ...  SAS  CAS  MDE  CAH  total  imputation_panel  \\\n",
       "26           0           0  ...    0    0    0    0      0                 0   \n",
       "98           0           0  ...    0    0    0    0      0                 0   \n",
       "99           0           0  ...    0    0    0    0      0                 0   \n",
       "100          0           0  ...    0    0    0    0      0                 0   \n",
       "112          0           0  ...    0    0    0    0      0                 0   \n",
       "115          0           0  ...    0    0    0    0      0                 0   \n",
       "116          0           0  ...    0    0    0    0      0                 0   \n",
       "156          0           0  ...    0    0    0    0      0                 0   \n",
       "158          0           0  ...    0    0    0    0      0                 0   \n",
       "162          0           0  ...    0    0    0    0      0                 0   \n",
       "165          0           0  ...    0    0    0    0      0                 0   \n",
       "180          0           0  ...    0    0    0    0      0                 0   \n",
       "181          0           0  ...    0    0    0    0      0                 0   \n",
       "185          0           0  ...    0    0    0    0      0                 0   \n",
       "190          0           0  ...    0    0    0    0     22                 0   \n",
       "210          0           0  ...    0    0    0    0      0                 0   \n",
       "233          0           0  ...    0    0    0    0      0                 0   \n",
       "269          0           0  ...    0    0    0    0      0                 0   \n",
       "276          0           0  ...    0    0    0    0      0                 0   \n",
       "\n",
       "     imputation_complete  genotypes_shareable  gdpr         site  \n",
       "26                     0                    0     1            0  \n",
       "98                     0                    0     0         DZNE  \n",
       "99                     0                    0     0            0  \n",
       "100                    0                    1     0            0  \n",
       "112                    0                    0     0       Lubeck  \n",
       "115                    0                    0     0            0  \n",
       "116                    0                    0     0            0  \n",
       "156                    0                    0     0         DZNE  \n",
       "158                    0                    0     0            0  \n",
       "162                    0                    0     0            0  \n",
       "165                    0                    0     0            0  \n",
       "180                    0                    0     0  NIH, Lubeck  \n",
       "181                    0                    0     0       Lubeck  \n",
       "185                    0                    0     0            0  \n",
       "190                    0                    0     0         DZNE  \n",
       "210                    0                    0     0            0  \n",
       "233                    0                    0     0       Lubeck  \n",
       "269                    0                    0     0            0  \n",
       "276                    0                    0     0            0  \n",
       "\n",
       "[19 rows x 27 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out.loc[df_out.study_code=='0']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
