{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import difflib\n",
    "from sqlalchemy import create_engine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gt_df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "gmmb_df = pd.read_csv('data/gmmb_tracker.csv', header=1)\n",
    "ancestry_df = pd.read_csv('data/FEB_2024_cohort_ancestry_phenotype.csv', header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Date</th>\n",
       "      <th>Updater</th>\n",
       "      <th>Study Code</th>\n",
       "      <th>Individual Study Name</th>\n",
       "      <th>Institution</th>\n",
       "      <th>Monogenic / Complex / Mixed</th>\n",
       "      <th>if Mixed: estimated proportion in Monogenic</th>\n",
       "      <th>City</th>\n",
       "      <th>Geographic locality [free text]</th>\n",
       "      <th>PI</th>\n",
       "      <th>...</th>\n",
       "      <th>Predicted # DLB by 2028</th>\n",
       "      <th>Predicted # CBS by 2028</th>\n",
       "      <th>Predicted # Prodromal by 2028</th>\n",
       "      <th>Predicted # Other atypical Parkinsonism cases by 2028</th>\n",
       "      <th>Cohort to be WGSd\\nYes = 1\\nNo = 0</th>\n",
       "      <th>WGS important notes</th>\n",
       "      <th>Sequenced samples [N]</th>\n",
       "      <th>Samples passed WGS QC [N]</th>\n",
       "      <th>Shared to AMP-PD [0 = no, 1 = yes].1</th>\n",
       "      <th>Notes [free text]</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2021-02-09</td>\n",
       "      <td>Hirotaka.I</td>\n",
       "      <td>BCM</td>\n",
       "      <td>Baylor College of Medicine (BCM)</td>\n",
       "      <td>Baylor College of Medicine</td>\n",
       "      <td>Complex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Houston,TX</td>\n",
       "      <td>USA</td>\n",
       "      <td>Josh Shulman</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2021-02-09</td>\n",
       "      <td>Hirotaka.I</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Complex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Camden, NJ</td>\n",
       "      <td>USA</td>\n",
       "      <td>Cornelis Blauwendraat</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2021-02-09</td>\n",
       "      <td>Hirotaka.I</td>\n",
       "      <td>PDGNRTN</td>\n",
       "      <td>PDGENEration</td>\n",
       "      <td>Parkinson's Foundation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Miami, FL</td>\n",
       "      <td>USA</td>\n",
       "      <td>Roy Alcalay</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>255.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2021-02-09</td>\n",
       "      <td>Hirotaka.I</td>\n",
       "      <td>UMD</td>\n",
       "      <td>UMD</td>\n",
       "      <td>University of Maryland</td>\n",
       "      <td>Complex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Baltimore, MD</td>\n",
       "      <td>USA</td>\n",
       "      <td>Lisa Shulman</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2021-03-03</td>\n",
       "      <td>Hirotaka.I</td>\n",
       "      <td>PROSPOS</td>\n",
       "      <td>PROSPOS</td>\n",
       "      <td>Oslo University Hospital</td>\n",
       "      <td>Complex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Oslo</td>\n",
       "      <td>Norway</td>\n",
       "      <td>Lasse Pihlstrøm</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>2024-05-08</td>\n",
       "      <td>Simona Jasaityte</td>\n",
       "      <td>FMGC</td>\n",
       "      <td>French MSA genetic cohort</td>\n",
       "      <td>Bordeaux University Hospital</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>Wassilios Meissner</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>2024-05-08</td>\n",
       "      <td>Simona Jasaityte</td>\n",
       "      <td>IRAMS</td>\n",
       "      <td>Etude de l’insulino-résistance dans l’atrophie...</td>\n",
       "      <td>Bordeaux University Hospital</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>Wassilios Meissner</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>2024-05-08</td>\n",
       "      <td>Simona Jasaityte</td>\n",
       "      <td>GENOT</td>\n",
       "      <td>Genetic factors associated with the penetrance...</td>\n",
       "      <td>Department of Neurology, Razi Hospital, Tunis;...</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tunis</td>\n",
       "      <td>Tunisia</td>\n",
       "      <td>Riadh Gouider</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>2024-05-16</td>\n",
       "      <td>Simona Jasaityte</td>\n",
       "      <td>UTSW</td>\n",
       "      <td>Understanding the Genetic link for Parkinson's...</td>\n",
       "      <td>UT Southwestern Medical Center</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dallas</td>\n",
       "      <td>United States</td>\n",
       "      <td>Vibhash Sharma</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>2024-05-16</td>\n",
       "      <td>Simona Jasaityte</td>\n",
       "      <td>ADPAR</td>\n",
       "      <td>Investigation of the genome of patients with a...</td>\n",
       "      <td>Carol Davila University of Medicine and Pharma...</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bucharest</td>\n",
       "      <td>Romania</td>\n",
       "      <td>Bogdan Ovidiu Popescu</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Date           Updater Study Code  \\\n",
       "0    2021-02-09        Hirotaka.I        BCM   \n",
       "1    2021-02-09        Hirotaka.I    CORIELL   \n",
       "2    2021-02-09        Hirotaka.I    PDGNRTN   \n",
       "3    2021-02-09        Hirotaka.I        UMD   \n",
       "4    2021-03-03        Hirotaka.I    PROSPOS   \n",
       "..          ...               ...        ...   \n",
       "306  2024-05-08  Simona Jasaityte       FMGC   \n",
       "307  2024-05-08  Simona Jasaityte      IRAMS   \n",
       "308  2024-05-08  Simona Jasaityte      GENOT   \n",
       "309  2024-05-16  Simona Jasaityte       UTSW   \n",
       "310  2024-05-16  Simona Jasaityte      ADPAR   \n",
       "\n",
       "                                 Individual Study Name  \\\n",
       "0                     Baylor College of Medicine (BCM)   \n",
       "1                                              CORIELL   \n",
       "2                                         PDGENEration   \n",
       "3                                                  UMD   \n",
       "4                                              PROSPOS   \n",
       "..                                                 ...   \n",
       "306                          French MSA genetic cohort   \n",
       "307  Etude de l’insulino-résistance dans l’atrophie...   \n",
       "308  Genetic factors associated with the penetrance...   \n",
       "309  Understanding the Genetic link for Parkinson's...   \n",
       "310  Investigation of the genome of patients with a...   \n",
       "\n",
       "                                           Institution  \\\n",
       "0                           Baylor College of Medicine   \n",
       "1                                                  NaN   \n",
       "2                               Parkinson's Foundation   \n",
       "3                               University of Maryland   \n",
       "4                             Oslo University Hospital   \n",
       "..                                                 ...   \n",
       "306                       Bordeaux University Hospital   \n",
       "307                       Bordeaux University Hospital   \n",
       "308  Department of Neurology, Razi Hospital, Tunis;...   \n",
       "309                     UT Southwestern Medical Center   \n",
       "310  Carol Davila University of Medicine and Pharma...   \n",
       "\n",
       "    Monogenic / Complex / Mixed  if Mixed: estimated proportion in Monogenic  \\\n",
       "0                       Complex                                          NaN   \n",
       "1                       Complex                                          NaN   \n",
       "2                         Mixed                                          NaN   \n",
       "3                       Complex                                          NaN   \n",
       "4                       Complex                                          NaN   \n",
       "..                          ...                                          ...   \n",
       "306                       Mixed                                          NaN   \n",
       "307                       Mixed                                          NaN   \n",
       "308                       Mixed                                          NaN   \n",
       "309                       Mixed                                          NaN   \n",
       "310                       Mixed                                          NaN   \n",
       "\n",
       "              City Geographic locality [free text]                     PI  \\\n",
       "0       Houston,TX                             USA           Josh Shulman   \n",
       "1       Camden, NJ                             USA  Cornelis Blauwendraat   \n",
       "2       Miami, FL                              USA            Roy Alcalay   \n",
       "3    Baltimore, MD                             USA           Lisa Shulman   \n",
       "4             Oslo                          Norway        Lasse Pihlstrøm   \n",
       "..             ...                             ...                    ...   \n",
       "306       Bordeaux                          France     Wassilios Meissner   \n",
       "307       Bordeaux                          France     Wassilios Meissner   \n",
       "308          Tunis                         Tunisia          Riadh Gouider   \n",
       "309         Dallas                   United States         Vibhash Sharma   \n",
       "310      Bucharest                         Romania  Bogdan Ovidiu Popescu   \n",
       "\n",
       "     ... Predicted # DLB by 2028 Predicted # CBS by 2028  \\\n",
       "0    ...                     NaN                     NaN   \n",
       "1    ...                     NaN                     NaN   \n",
       "2    ...                     NaN                     NaN   \n",
       "3    ...                     NaN                     NaN   \n",
       "4    ...                     NaN                     NaN   \n",
       "..   ...                     ...                     ...   \n",
       "306  ...                     0.0                     0.0   \n",
       "307  ...                     0.0                     0.0   \n",
       "308  ...                     0.0                     0.0   \n",
       "309  ...                     0.0                     0.0   \n",
       "310  ...                     0.0                     0.0   \n",
       "\n",
       "    Predicted # Prodromal by 2028  \\\n",
       "0                             NaN   \n",
       "1                             NaN   \n",
       "2                             NaN   \n",
       "3                             NaN   \n",
       "4                             NaN   \n",
       "..                            ...   \n",
       "306                           0.0   \n",
       "307                           0.0   \n",
       "308                           0.0   \n",
       "309                           0.0   \n",
       "310                           0.0   \n",
       "\n",
       "    Predicted # Other atypical Parkinsonism cases by 2028  \\\n",
       "0                                                  NaN      \n",
       "1                                                  NaN      \n",
       "2                                                  NaN      \n",
       "3                                                  NaN      \n",
       "4                                                  NaN      \n",
       "..                                                 ...      \n",
       "306                                                0.0      \n",
       "307                                                0.0      \n",
       "308                                                0.0      \n",
       "309                                                0.0      \n",
       "310                                                0.0      \n",
       "\n",
       "    Cohort to be WGSd\\nYes = 1\\nNo = 0 WGS important notes  \\\n",
       "0                                  NaN                 NaN   \n",
       "1                                  NaN                 NaN   \n",
       "2                                  NaN                 NaN   \n",
       "3                                  NaN                 NaN   \n",
       "4                                  NaN                 NaN   \n",
       "..                                 ...                 ...   \n",
       "306                                1.0                 NaN   \n",
       "307                                1.0                 NaN   \n",
       "308                                1.0                 NaN   \n",
       "309                                NaN                 NaN   \n",
       "310                                NaN                 NaN   \n",
       "\n",
       "    Sequenced samples [N]  Samples passed WGS QC [N]  \\\n",
       "0                      NaN                       NaN   \n",
       "1                      NaN                       NaN   \n",
       "2                    255.0                     235.0   \n",
       "3                      NaN                       NaN   \n",
       "4                      NaN                       NaN   \n",
       "..                     ...                       ...   \n",
       "306                    NaN                       NaN   \n",
       "307                    NaN                       NaN   \n",
       "308                    NaN                       NaN   \n",
       "309                    NaN                       NaN   \n",
       "310                    NaN                       NaN   \n",
       "\n",
       "    Shared to AMP-PD [0 = no, 1 = yes].1  Notes [free text]  \n",
       "0                                    1.0                NaN  \n",
       "1                                    1.0                NaN  \n",
       "2                                    1.0                NaN  \n",
       "3                                    1.0                NaN  \n",
       "4                                    NaN                NaN  \n",
       "..                                   ...                ...  \n",
       "306                                  NaN                NaN  \n",
       "307                                  NaN                NaN  \n",
       "308                                  NaN                NaN  \n",
       "309                                  NaN                NaN  \n",
       "310                                  NaN                NaN  \n",
       "\n",
       "[311 rows x 69 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gt_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## match gp2 'Study Code' to gmmb 'Cohort Name'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# def find_best_matches(list1, list2, cutoff=0.6):\n",
    "#     matches = {}\n",
    "#     for item in list1:\n",
    "#         match = difflib.get_close_matches(item, list2, n=1, cutoff=cutoff)\n",
    "#         if match:\n",
    "#             matches[item] = match[0]\n",
    "#         else:\n",
    "#             matches[item] = None\n",
    "#     return matches\n",
    "\n",
    "# best_matches = find_best_matches(gt_df.loc[~gt_df['Study Code'].isna(), 'Study Code'], gmmb_df['Cohort Name'])\n",
    "\n",
    "# for k, v in best_matches.items():\n",
    "#     print(f\"{k}: {v}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genotracker\n",
      "Study Code: ANDPD\n",
      "Individual Study Name: Anxiety in Parkinson's disease\n",
      "Institution: University College London\n",
      "City: London\n",
      "Geographic locality [free text]: United Kingdom\n",
      "\n",
      "############################################################\n",
      "\n",
      "gmmb\n",
      "############################################################\n",
      "\n"
     ]
    }
   ],
   "source": [
    "study_code = \"ANDPD\"\n",
    "\n",
    "gt_fields = ['Study Code', 'Individual Study Name', 'Institution', 'City', 'Geographic locality [free text]']\n",
    "# gt_fields_df = gt_df.loc[gt_df['Study Code']==study_code, gt_fields]\n",
    "gt_fields_df = gt_df.loc[gt_df['Institution'].str.contains(\"National Institute of Mental Health\", na=False), gt_fields]\n",
    "\n",
    "gmmb_fields = ['Cohort Name', 'Institution', 'City', 'Country (Location)']\n",
    "gmmb_fields_df = gmmb_df.loc[gmmb_df[\"Institution\"].str.contains(\"UCL\"), gmmb_fields]\n",
    "\n",
    "\n",
    "print('genotracker')\n",
    "for index, row in gt_fields_df.iterrows():\n",
    "    for col_name, value in row.items():\n",
    "        print(f'{col_name}: {value}')\n",
    "    print()\n",
    "\n",
    "print('##'*30)\n",
    "print()\n",
    "\n",
    "print('gmmb')\n",
    "for index, row in gmmb_fields_df.iterrows():\n",
    "    for col_name, value in row.items():\n",
    "        print(f'{col_name}: {value}')\n",
    "    print()\n",
    "    \n",
    "print('##'*30)\n",
    "print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 726,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "167"
      ]
     },
     "execution_count": 726,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# gmmb_df['Name on Ancestry Phenotype Backup']\n",
    "# gt_df.merge(gmmb_df, how='left', left_on='Study Code', right_on='Name on Ancestry Phenotype Backup')\n",
    "matching_cohorts = {\n",
    "    \"BCM\": \"Baylor College of Medicine (BCM)\",\n",
    "    \"CORIELL\": \"CORIELL\",\n",
    "    \"PDGNRTN\": \"PDGENEration\",\n",
    "    \"UMD\": \"BCM-UMD\",\n",
    "    \"PROSPOS\": \"PROSPOS\",\n",
    "    \"SP-SEV\": \"Sp_SEV - HUVR/CSIC/Universidad de Sevill\",\n",
    "    \"COPN\": \"C-OPN\",\n",
    "    \"NZP3\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"UPENNU19\": \"UPENN U19 Cohort\",\n",
    "    \"BADGEPD\": \"BADGE-PD\",\n",
    "    \"CAMPAIGN\": \"CamPaIGN\",\n",
    "    \"DIGPD\": \"DIGPD\",\n",
    "    \"ICEBERG\": \"ICEBERG\",\n",
    "    \"IDCME\": \"NGC\",\n",
    "    \"MIND\": \"MIND Initiative\",\n",
    "    \"NSPARK\": \"NS PARK\",\n",
    "    \"PICNICS\": \"PICNICS\",\n",
    "    \"PREDISTIM\": \"PREDICT-PD\",\n",
    "    \"QPN\": \"Quebec Parkinson's Network\",\n",
    "    \"RBM\": \"RBM\",\n",
    "    \"SP-BCN3\": \"Sp_BCN3 - Sant Joan de Déu\",\n",
    "    \"SP-OVI\": \"Sp_OVI - University of Oviedo\",\n",
    "    \"MDGAP-KINGS\": \"MD GAP - King's College London\",\n",
    "    \"MDGAP-QSBB\": \"MD GAP - QSBB - University College London\",\n",
    "    \"POM\": \"POM\",\n",
    "    \"LCC\": \"LCC\",\n",
    "    \"UMKLM\": \"Malaysian Parkinson's Genetics Cohort - UMKLM\",\n",
    "    \"GPARKGR\": \"GenePARKGR\",\n",
    "    \"PROPARK\": \"ProPark\",\n",
    "    \"SP-BCN2\": \"Sp_BCN2 - University Hospital Mútua de Terrassa\",\n",
    "    \"SP-SNT\": \"Sp_SNT - University Hospital Marques de Valdecilla-IDIVAL\",\n",
    "    \"BIOFIND\": \"BioFIND\",\n",
    "    \"BLAACPD-KPM\": \"BLAACPD - Kaiser Permanente\",\n",
    "    \"BLAACPD-RUSH\": \"BLAACPD - Rush University\",\n",
    "    \"BLAACPD-UAB\": \"BLAACPD - University of Alabama at Birmingham\",\n",
    "    \"BLAACPD-UC\": \"BLAACPD - University of Chicago\",\n",
    "    \"CALYPSO\": \"CALYPSO\",\n",
    "    \"MDGAP-EBB\": \"MD GAP - Edinburgh Brain Bank - University of Edinburgh\",\n",
    "    \"PFP\": \"PFP\",\n",
    "    \"PPMI\": \"PPMI\",\n",
    "    \"PROBAND\": \"PROBAND\",\n",
    "    \"SYNAPS-KZ\": \"SYNAPS-KZ\",\n",
    "    \"VIPD\": \"VIPD\",\n",
    "    \"CANDAS\": \"CANDAS\",\n",
    "    \"QPP\": \"QPP\",\n",
    "    \"UGUPD\": \"UGCPD\",\n",
    "    \"GENEFINDER\": \"GeneFinder\",\n",
    "    \"OPDC\": \"OPDC Discovery Cohort\",\n",
    "    \"MDGAP-MBB\": \"Manchester Brain Bank\",\n",
    "    \"PROMO\": \"ProtectMove - Institute of Neurogenetics, University of Lübeck\",\n",
    "    \"PROATYP\": \"proATYP\",\n",
    "    \"MDGAP-IBB\": \"PUKBB\",\n",
    "    \"BENHAMIDA\": \"Clinicogenetic Study of PD in Tunisia\",\n",
    "    \"ELPD\": \"ELPD Project\",\n",
    "    \"PDSGR\": \"PDS\",\n",
    "    \"PREDICTPD\": \"PREDICT-PD\",\n",
    "    \"AUTH\": \"AUTH - GP2 study\",\n",
    "    \"PDSTAT\": \"PD STAT\",\n",
    "    \"MPBC\": \"MPBC\",\n",
    "    \"SYDPDRC\": \"Sydney PDRC\",\n",
    "    \"APGS\": \"APGS\",\n",
    "    \"SYDBB\": \"Sydney Brain Bank - SYDBB\",\n",
    "    \"CPRC\": \"Cambridge Parkinson's Research Clinic\",\n",
    "    \"IPDGCEA-JKU\": \"IPDGC East Asia - Jikei University\",\n",
    "    \"ADEPTPD\": \"ADepT-PD\",\n",
    "    \"PDMDC\": \"PDMDC/NUgene\",\n",
    "    \"GARVAN\": \"Genetic analysis in inherited neurological disorders\",\n",
    "    \"KUL\": \"Genotype-Phenotype Correlations in PD and Related Movement Disorders - KUL\",\n",
    "    \"PDMDCB\": \"Parkinson Disease and Movement Disorders Center Biorepository\",\n",
    "    \"TUEPAC\": \"Tuepac + TREND\",\n",
    "    \"LUXPARK\": \"LUX-Park\",\n",
    "    \"BIOPARK\": \"To be updated - Karolinska Institutet\",\n",
    "    \"IMMLISBON\": \"Movement Disorders Biobanc\",\n",
    "    \"PDGEN\": \"PDGENEration\",\n",
    "    \"DATATOP\": \"DATATOP\",\n",
    "    \"SZTE\": \"SZTE\",\n",
    "    \"TRD\": \"Trondheim Study\",\n",
    "    \"MDGAP-SW\": \"MD GAP - South West Dementia Brain Bank - Southmead Hospital\",\n",
    "    \"S4\": \"S4\",\n",
    "    \"NETPDLS1\": \"NET-PD\",\n",
    "    \"DENOPA\": \"DeNoPa\",\n",
    "    \"NTUH\": \"Genomewide Association Study in PD among East Asians - NTUH\",\n",
    "    \"BRFAA\": \"Biological markers in Parkinson's disease - BRFAA\",\n",
    "    \"BUE\": \"BUE - GP2\",\n",
    "    \"LARGEPD\": \"To be updated - Cleveland Clinic\",\n",
    "    \"BBDP\": \"Arizona Brain Bank Brain and body donation programme (BBDP)\",\n",
    "    \"CINCINNATI\": \"Clinical and genetic study of Unusual/Familial Movement Disorders\",\n",
    "    \"MMCTU\": \"Genotype-Phenotype Correlations in Nepalese PD Population - MMCTU\",\n",
    "    \"PERUGIA\": \"CSF Biomarkers\",\n",
    "    \"STELLENBOS\": \"Genetic study of PD and related movement disorders\",\n",
    "    \"CHULALONGK\": \"Genetic Testing in Patients with Movement Disorders and PD\",\n",
    "    \"MDGAP-NWCSTL\": \"MD GAP - Newcastle Brain Bank - University of Newcastle\",\n",
    "    \"KFSH\": \"The Molecular Basis of Movement Disorders in Saudis - KFSH\",\n",
    "    \"PAGE\": \"PAGE\",\n",
    "    \"SYNAPS-GR\": \"SYNAPS-GR\",\n",
    "    \"PDCARE\": \"PDCARE\",\n",
    "    \"RAPSODI\": \"RAPSODI\",\n",
    "    \"YOPDFL\": \"YOPDFL (Young Onset Parkinson's Disease)\",\n",
    "    \"FDSANTIAGO\": \"GP2 - Fundación Diagnosis\",\n",
    "    \"MULODZ\": \"Molecular Testing of Polish PD Patients\",\n",
    "    \"TONIC-PERRON\": \"TONiC PD\",\n",
    "    \"KOC\": \"GP2 - Koc University\",\n",
    "    \"JGUM\": \"JGUM\",\n",
    "    \"PATRAS\": \"PATRAS\",\n",
    "    \"NUGENE\": \"NU Gene\",\n",
    "    \"CDPDGR\": \"Thessaly PD Database\",\n",
    "    \"JENPARK\": \"Jena Parkinson Study\",\n",
    "    \"UROMETV\": \"Standardized data collection of PD patients\",\n",
    "    \"CHCHPD\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"HIPDR\": \"HIPDR\",\n",
    "    \"KOLLING\": \"Kolling PD\",\n",
    "    \"SEOULNUH\": \"GP2 - Seoul National University Hospital\",\n",
    "    \"NROSF2U\": \"Multicentric Retrospective and Prospective PD Database\",\n",
    "    \"BLAACPD-LSU\": \"BLAACPD - Louisiana State University\",\n",
    "    \"IMMUNEPD\": \"Innate and Adaptive Immunity in PD\",\n",
    "    \"PROMO-EPIPARK\": \"PROMO-EPIPARK\",\n",
    "    \"PROMO-GENEPARK\": \"PROMO-GENEPARK\",\n",
    "    \"PROMO-ROPAD\": \"PROMO-ROPAD\",\n",
    "    \"PROMO-SFB134\": \"PROMO-SFB134\",\n",
    "    \"PROMO-SYSMED\": \"PROMO-SYSMEDPD\",\n",
    "    \"OXQUIP\": \"Oxford Study of Quantification in Parkinsonism\",\n",
    "    \"PROMO-USTP\": \"PROMO-USTP\",\n",
    "    \"MUVIENNA\": \"Families with Neurodegenerative Diseases\",\n",
    "    \"CATANZARO\": \"Familial and Early-onset PD Database\",\n",
    "    \"QMAP\": \"Quantitative MRI for Anatomical Phenotyping in Parkinson's Disease\",\n",
    "    \"CATPD\": \"Central Asian and Transcaucasian PD (CAT-PD)\",\n",
    "    \"IPDGCAF-NG\": \"IPDGC Africa - The Nigerian Parkinson Disease Research Network\",\n",
    "    \"IPDGCEA-JU\": \"IPDGC East Asia - Juntendo University\",\n",
    "    \"TUMBB\": \"Biobank for Diseases of the Peripheral and Central Nervous System - TUMBB\",\n",
    "    \"TUMPD\": \"Genetics of Parkinson's Disease - TUMPD\",\n",
    "    \"ANANEOS\": \"ANANEOS Study\",\n",
    "    \"MDGAP-CAMBRIDGE\": \"MD GAP - Cambridge\",\n",
    "    \"MDGAP-OXF\": \"MD GAP - OXF\",\n",
    "    \"VETPD\": \"VET-PD\",\n",
    "    \"BERNGEN\": 'BERN',\n",
    "    \"SINAI\": 'Mount Sinai Cohort',\n",
    "    \"PARKWEST\": \"The Norwegian Parkwest Study\",\n",
    "    \"PROSPECT\": \"PROSPECT\",\n",
    "    \"UMUDALL\": \"UMUDALL\",\n",
    "    \"TORONTO\": \"To be updated - University of Toronto\",\n",
    "    \"MDGAP-VBB\": \"MD GAP - Victorian Brain Bank\",\n",
    "    \"IRCCSN\": \"Genetics of Neurological Pathologies: New Variations, New Phenotypes\",\n",
    "    \"PROBEPD\": \"PROBEPD\",\n",
    "    \"TLVMC\": \"DLB Cohort TLVMC\",\n",
    "    \"ANDPD\": \"ANDPD\",    \n",
    "    \"MONPAR\": \"Genetic Study for Pakinson's Disease in Mongolia\",\n",
    "    \"NGIPDO\": \"NGIPDO - Neurogenomics and Informatics Center - Parkinson's Disease  Deep Molecular Profiling\",\n",
    "    \"IPDGCEA-UKM\": \"IPDGC East Asia - UKM\",\n",
    "    \"DODONA\": \"The DodoNa Project: DNA Predictions to Improve Neurological Health\",\n",
    "    \"PROMO-CARRIERS\": \"PROMO-CARRIERS\",\n",
    "    \"HAWAIIPD\": \"HAWAII PD\",\n",
    "    \"YMS\": \"Yale Medical School\",\n",
    "    \"IGECND\": \"IGECND\",\n",
    "    \"DHAKA\": \"GP2 - Bangladesh\",\n",
    "    \"UGCPD\": \"UGCPD\",\n",
    "    \"AAPDGC\": \"AfrAbia-PD Genomic Consortium (AA-PDGC)\",\n",
    "    \"REGARDS\": \"REasons for Geographic and Racial Differences in Stroke (REGARDS) Project\",\n",
    "    \"NZPEGS\": \"New Zealand Parkinson's Environment and Genes Study\",\n",
    "    \"GPX\": \"GenePool Study\",\n",
    "    \"VAIBB\": \"Van Andel Institute Brain Bank\",\n",
    "    \"WEHI\": \"WEHI - Integrating molecular and omics approaches in Parkinson’s disease\",\n",
    "    \"ACCESSPD\": \"ACCESSPD\",\n",
    "    \"GPS\": \"GenePool Study\",\n",
    "    \"NGP\": \"The Neuro Genomics Partnership\",\n",
    "    \"SCANPSP\": \"SCANPSP - Study of Comprehensive ANd Multimodal Marker-based Cohort of PSP\",\n",
    "    \"SAMSUNG\": \"GP2 - Samsung\",\n",
    "    \"BURGOS\": \"Local Study - Burgos\"\n",
    "}\n",
    "\n",
    "len(matching_cohorts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gt_name_merge = gt_df[['Study Code','Individual Study Name']].merge(gmmb_df[['Cohort Name']], left_on='Individual Study Name', right_on='Cohort Name')\n",
    "# for index, row in gt_name_merge.iterrows():\n",
    "#     print(f'Row {index}')\n",
    "#     for col_name, value in row.items():\n",
    "#         print(f\" {col_name}: {value}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_mapping = {\n",
    "    \"BCM\": \"Baylor College of Medicine (BCM)\",\n",
    "    \"CORIELL\": \"CORIELL\",\n",
    "    \"PDGNRTN\": \"PDGENEration\",\n",
    "    \"UMD\": \"BCM-UMD\",\n",
    "    \"PROSPOS\": \"PROSPOS\",\n",
    "    \"SP-SEV\": \"Sp_SEV - HUVR/CSIC/Universidad de Sevill\",\n",
    "    \"COPN\": \"C-OPN\",\n",
    "    \"NZP3\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"UPENNU19\": \"UPENN U19 Cohort\",\n",
    "    \"BADGEPD\": \"BADGE-PD\",\n",
    "    \"CAMPAIGN\": \"CamPaIGN\",\n",
    "    \"DIGPD\": \"DIGPD\",\n",
    "    \"ICEBERG\": \"ICEBERG\",\n",
    "    \"IDCME\": \"NGC\",\n",
    "    \"MIND\": \"MIND Initiative\",\n",
    "    \"NSPARK\": \"NS PARK\",\n",
    "    \"PICNICS\": \"PICNICS\",\n",
    "    \"PREDISTIM\": \"PREDICT-PD\",\n",
    "    \"QPN\": \"Quebec Parkinson's Network\",\n",
    "    \"RBM\": \"RBM\",\n",
    "    \"SP-BCN3\": \"Sp_BCN3 - Sant Joan de Déu\",\n",
    "    \"SP-OVI\": \"Sp_OVI - University of Oviedo\",\n",
    "    \"MDGAP-KINGS\": \"MD GAP - King's College London\",\n",
    "    \"MDGAP-QSBB\": \"MD GAP - QSBB - University College London\",\n",
    "    \"POM\": \"POM\",\n",
    "    \"LCC\": \"LCC\",\n",
    "    \"UMKLM\": \"Malaysian Parkinson's Genetics Cohort - UMKLM\",\n",
    "    \"GPARKGR\": \"GenePARKGR\",\n",
    "    \"PROPARK\": \"ProPark\",\n",
    "    \"SP-BCN2\": \"Sp_BCN2 - University Hospital Mútua de Terrassa\",\n",
    "    \"SP-SNT\": \"Sp_SNT - University Hospital Marques de Valdecilla-IDIVAL\",\n",
    "    \"BIOFIND\": \"BioFIND\",\n",
    "    \"BLAACPD-KPM\": \"BLAACPD - Kaiser Permanente\",\n",
    "    \"BLAACPD-RUSH\": \"BLAACPD - Rush University\",\n",
    "    \"BLAACPD-UAB\": \"BLAACPD - University of Alabama at Birmingham\",\n",
    "    \"BLAACPD-UC\": \"BLAACPD - University of Chicago\",\n",
    "    \"CALYPSO\": \"CALYPSO\",\n",
    "    \"MDGAP-EBB\": \"MD GAP - Edinburgh Brain Bank - University of Edinburgh\",\n",
    "    \"PFP\": \"PFP\",\n",
    "    \"PPMI\": \"PPMI\",\n",
    "    \"PROBAND\": \"PROBAND\",\n",
    "    \"SYNAPS-KZ\": \"SYNAPS-KZ\",\n",
    "    \"VIPD\": \"VIPD\",\n",
    "    \"CANDAS\": \"CANDAS\",\n",
    "    \"QPP\": \"QPP\",\n",
    "    \"UGUPD\": \"UGCPD\",\n",
    "    \"GENEFINDER\": \"GeneFinder\",\n",
    "    \"OPDC\": \"OPDC Discovery Cohort\",\n",
    "    \"MDGAP-MBB\": \"Manchester Brain Bank\",\n",
    "    \"PROMO\": \"ProtectMove - Institute of Neurogenetics, University of Lübeck\",\n",
    "    \"PROATYP\": \"proATYP\",\n",
    "    \"MDGAP-IBB\": \"PUKBB\",\n",
    "    \"BENHAMIDA\": \"Clinicogenetic Study of PD in Tunisia\",\n",
    "    \"ELPD\": \"ELPD Project\",\n",
    "    \"PDSGR\": \"PDS\",\n",
    "    \"PREDICTPD\": \"PREDICT-PD\",\n",
    "    \"AUTH\": \"AUTH - GP2 study\",\n",
    "    \"PDSTAT\": \"PD STAT\",\n",
    "    \"MPBC\": \"MPBC\",\n",
    "    \"SYDPDRC\": \"Sydney PDRC\",\n",
    "    \"APGS\": \"APGS\",\n",
    "    \"SYDBB\": \"Sydney Brain Bank - SYDBB\",\n",
    "    \"CPRC\": \"Cambridge Parkinson's Research Clinic\",\n",
    "    \"IPDGCEA-JKU\": \"IPDGC East Asia - Jikei University\",\n",
    "    \"ADEPTPD\": \"ADepT-PD\",\n",
    "    \"PDMDC\": \"PDMDC/NUgene\",\n",
    "    \"GARVAN\": \"Genetic analysis in inherited neurological disorders\",\n",
    "    \"KUL\": \"Genotype-Phenotype Correlations in PD and Related Movement Disorders - KUL\",\n",
    "    \"PDMDCB\": \"Parkinson Disease and Movement Disorders Center Biorepository\",\n",
    "    \"TUEPAC\": \"Tuepac + TREND\",\n",
    "    \"LUXPARK\": \"LUX-Park\",\n",
    "    \"BIOPARK\": \"To be updated - Karolinska Institutet\",\n",
    "    \"IMMLISBON\": \"Movement Disorders Biobanc\",\n",
    "    \"PDGEN\": \"PDGENEration\",\n",
    "    \"DATATOP\": \"DATATOP\",\n",
    "    \"SZTE\": \"SZTE\",\n",
    "    \"TRD\": \"Trondheim Study\",\n",
    "    \"MDGAP-SW\": \"MD GAP - South West Dementia Brain Bank - Southmead Hospital\",\n",
    "    \"S4\": \"S4\",\n",
    "    \"NETPDLS1\": \"NET-PD\",\n",
    "    \"DENOPA\": \"DeNoPa\",\n",
    "    \"NTUH\": \"Genomewide Association Study in PD among East Asians - NTUH\",\n",
    "    \"BRFAA\": \"Biological markers in Parkinson's disease - BRFAA\",\n",
    "    \"BUE\": \"BUE - GP2\",\n",
    "    \"LARGEPD\": \"To be updated - Cleveland Clinic\",\n",
    "    \"BBDP\": \"Arizona Brain Bank Brain and body donation programme (BBDP)\",\n",
    "    \"CINCINNATI\": \"Clinical and genetic study of Unusual/Familial Movement Disorders\",\n",
    "    \"MMCTU\": \"Genotype-Phenotype Correlations in Nepalese PD Population - MMCTU\",\n",
    "    \"PERUGIA\": \"CSF Biomarkers\",\n",
    "    \"STELLENBOS\": \"Genetic study of PD and related movement disorders\",\n",
    "    \"CHULALONGK\": \"Genetic Testing in Patients with Movement Disorders and PD\",\n",
    "    \"MDGAP-NWCSTL\": \"MD GAP - Newcastle Brain Bank - University of Newcastle\",\n",
    "    \"KFSH\": \"The Molecular Basis of Movement Disorders in Saudis - KFSH\",\n",
    "    \"PAGE\": \"PAGE\",\n",
    "    \"SYNAPS-GR\": \"SYNAPS-GR\",\n",
    "    \"PDCARE\": \"PDCARE\",\n",
    "    \"RAPSODI\": \"RAPSODI\",\n",
    "    \"YOPDFL\": \"YOPDFL (Young Onset Parkinson's Disease)\",\n",
    "    \"FDSANTIAGO\": \"GP2 - Fundación Diagnosis\",\n",
    "    \"MULODZ\": \"Molecular Testing of Polish PD Patients\",\n",
    "    \"TONIC-PERRON\": \"TONiC PD\",\n",
    "    \"KOC\": \"GP2 - Koc University\",\n",
    "    \"JGUM\": \"JGUM\",\n",
    "    \"PATRAS\": \"PATRAS\",\n",
    "    \"NUGENE\": \"NU Gene\",\n",
    "    \"CDPDGR\": \"Thessaly PD Database\",\n",
    "    \"JENPARK\": \"Jena Parkinson Study\",\n",
    "    \"UROMETV\": \"Standardized data collection of PD patients\",\n",
    "    \"CHCHPD\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"HIPDR\": \"HIPDR\",\n",
    "    \"KOLLING\": \"Kolling PD\",\n",
    "    \"SEOULNUH\": \"GP2 - Seoul National University Hospital\",\n",
    "    \"NROSF2U\": \"Multicentric Retrospective and Prospective PD Database\",\n",
    "    \"BLAACPD-LSU\": \"BLAACPD - Louisiana State University\",\n",
    "    \"IMMUNEPD\": \"Innate and Adaptive Immunity in PD\",\n",
    "    \"PROMO-EPIPARK\": \"PROMO-EPIPARK\",\n",
    "    \"PROMO-GENEPARK\": \"PROMO-GENEPARK\",\n",
    "    \"PROMO-ROPAD\": \"PROMO-ROPAD\",\n",
    "    \"PROMO-SFB134\": \"PROMO-SFB134\",\n",
    "    \"PROMO-SYSMED\": \"PROMO-SYSMEDPD\",\n",
    "    \"OXQUIP\": \"Oxford Study of Quantification in Parkinsonism\",\n",
    "    \"PROMO-USTP\": \"PROMO-USTP\",\n",
    "    \"MUVIENNA\": \"Families with Neurodegenerative Diseases\",\n",
    "    \"CATANZARO\": \"Familial and Early-onset PD Database\",\n",
    "    \"QMAP\": \"Quantitative MRI for Anatomical Phenotyping in Parkinson's Disease\",\n",
    "    \"CATPD\": \"Central Asian and Transcaucasian PD (CAT-PD)\",\n",
    "    \"IPDGCAF-NG\": \"IPDGC Africa - The Nigerian Parkinson Disease Research Network\",\n",
    "    \"IPDGCEA-JU\": \"IPDGC East Asia - Juntendo University\",\n",
    "    \"TUMBB\": \"Biobank for Diseases of the Peripheral and Central Nervous System - TUMBB\",\n",
    "    \"TUMPD\": \"Genetics of Parkinson's Disease - TUMPD\",\n",
    "    \"ANANEOS\": \"ANANEOS Study\",\n",
    "    \"MDGAP-CAMBRIDGE\": \"MD GAP - Cambridge\",\n",
    "    \"MDGAP-OXF\": \"MD GAP - OXF\",\n",
    "    \"VETPD\": \"VET-PD\",\n",
    "    \"BERNGEN\": 'BERN',\n",
    "    \"SINAI\": 'Mount Sinai Cohort',\n",
    "    \"PARKWEST\": \"The Norwegian Parkwest Study\",\n",
    "    \"PROSPECT\": \"PROSPECT\",\n",
    "    \"UMUDALL\": \"UMUDALL\",\n",
    "    \"TORONTO\": \"To be updated - University of Toronto\",\n",
    "    \"MDGAP-VBB\": \"MD GAP - Victorian Brain Bank\",\n",
    "    \"IRCCSN\": \"Genetics of Neurological Pathologies: New Variations, New Phenotypes\",\n",
    "    \"PROBEPD\": \"PROBEPD\",\n",
    "    \"TLVMC\": \"DLB Cohort TLVMC\",\n",
    "    \"ANDPD\": \"ANDPD\",    \n",
    "    \"MONPAR\": \"Genetic Study for Pakinson's Disease in Mongolia\",\n",
    "    \"NGIPDO\": \"NGIPDO - Neurogenomics and Informatics Center - Parkinson's Disease  Deep Molecular Profiling\",\n",
    "    \"IPDGCEA-UKM\": \"IPDGC East Asia - UKM\",\n",
    "    \"DODONA\": \"The DodoNa Project: DNA Predictions to Improve Neurological Health\",\n",
    "    \"PROMO-CARRIERS\": \"PROMO-CARRIERS\",\n",
    "    \"HAWAIIPD\": \"HAWAII PD\",\n",
    "    \"YMS\": \"Yale Medical School\",\n",
    "    \"IGECND\": \"IGECND\",\n",
    "    \"DHAKA\": \"GP2 - Bangladesh\",\n",
    "    \"UGCPD\": \"UGCPD\",\n",
    "    \"AAPDGC\": \"AfrAbia-PD Genomic Consortium (AA-PDGC)\",\n",
    "    \"REGARDS\": \"REasons for Geographic and Racial Differences in Stroke (REGARDS) Project\",\n",
    "    \"NZPEGS\": \"New Zealand Parkinson's Environment and Genes Study\",\n",
    "    \"GPX\": \"GenePool Study\",\n",
    "    \"VAIBB\": \"Van Andel Institute Brain Bank\",\n",
    "    \"WEHI\": \"WEHI - Integrating molecular and omics approaches in Parkinson’s disease\",\n",
    "    \"ACCESSPD\": \"ACCESSPD\",\n",
    "    \"GPS\": \"GenePool Study\",\n",
    "    \"NGP\": \"The Neuro Genomics Partnership\",\n",
    "    \"SCANPSP\": \"SCANPSP - Study of Comprehensive ANd Multimodal Marker-based Cohort of PSP\",\n",
    "    \"SAMSUNG\": \"GP2 - Samsung\",\n",
    "    \"BURGOS\": \"Local Study - Burgos\",    \n",
    "    \"SP-DON\": None,\n",
    "    \"ECMPDC\": None,\n",
    "    \"SP-GRANADA\": None,\n",
    "    \"AHEPA\": None,\n",
    "    \"CIDR\": None,\n",
    "    \"INKOSI\": None,\n",
    "    \"SP-BCN1\": None,\n",
    "    \"SANGERARDO\": None,\n",
    "    \"GEMELLI\": None,\n",
    "    \"NMDAT\": None,\n",
    "    \"TARTU\": None,\n",
    "    \"TEHRAN\": None,\n",
    "    \"KIEL\": None,\n",
    "    \"YAGUE\": None,\n",
    "    \"MAYO\": None,\n",
    "    \"AMSBB\": None,\n",
    "    \"PDSA\": None,\n",
    "    \"RDIPD\": None,\n",
    "    \"ANKARA\": None,\n",
    "    \"DUPARC\": None,\n",
    "    \"ESTBB\": None,\n",
    "    \"PUJBC\": None,\n",
    "    \"CMC\": None,\n",
    "    \"SAREMA\": None,\n",
    "    \"IPDGCEA-MU\": None,\n",
    "    \"AARHUS\": None,\n",
    "    \"CEGEMOD\": None,\n",
    "    \"JUNT\": None,\n",
    "    \"HSPH\": None,\n",
    "    \"NGRC\": None,\n",
    "    \"SILESIA\": None,\n",
    "    \"KRAKOW\": None,\n",
    "    \"BRNO\": None,\n",
    "    \"COLOMBO\": None,\n",
    "    \"ADRC\": None,\n",
    "    \"SILAD\": None,\n",
    "    \"STANPD\": None,\n",
    "    \"GENNIS\": None,\n",
    "    \"AHS\": None,\n",
    "    \"IDIBAPS\": None,\n",
    "    \"FLENI\": None,\n",
    "    \"EPIC\": None,\n",
    "    \"CCFPD\": None,\n",
    "    \"GRACE\": None,\n",
    "    \"NYSCF\": None,\n",
    "    \"PARS\": None,\n",
    "    \"PEG\": None,\n",
    "    \"SAMC\": None,\n",
    "    \"SAUD\": None,\n",
    "    \"COBB\": None,\n",
    "    \"PDCGC\": None,\n",
    "    \"SANTPAU\": None,\n",
    "    \"TRAPCAF\": None,\n",
    "    \"NITHAI\": None,\n",
    "    \"RTGRS\": None,\n",
    "    \"PROGENI\": None,\n",
    "    \"BIOPD\": None,\n",
    "    \"BOLOGNA\": None,\n",
    "    \"BPS\": None,\n",
    "    \"CABBS\": None,\n",
    "    \"CLGN\": None,\n",
    "    \"CUHK\": None,\n",
    "    \"GENCI\": None,\n",
    "    \"HEBA-BRC\": None,\n",
    "    \"HEBA-INNS\": None,\n",
    "    \"HEBA-KASSEL\": None,\n",
    "    \"INECO\": None,\n",
    "    \"INNS\": None,\n",
    "    \"MONT\": None,\n",
    "    \"NORTHSHORE\": None,\n",
    "    \"PARAGEN\": None,\n",
    "    \"FAIRPARK\": None,\n",
    "    \"PROLBD\": None,\n",
    "    \"RIMU\": None,\n",
    "    \"PARAGEN-MON\": None,\n",
    "    \"UCIPARK\": None,\n",
    "    \"GBA1NL\": None,\n",
    "    \"PDNEF\": None,\n",
    "    \"UCIPROS\": None,\n",
    "    \"NHGRI\": None,\n",
    "    \"CAGL\": None,\n",
    "    \"REMANT\": None,\n",
    "    \"REMEL\": None,\n",
    "    \"UDINE\": None,\n",
    "    \"REMEDIO\": None,\n",
    "    \"PAIR-PSP\": None,\n",
    "    \"PDBIOMARKERS\": None,\n",
    "    \"NAFPD\": None,\n",
    "    \"FNPREG\": None,\n",
    "    \"GED\": None,\n",
    "    \"PDGEN-CLEM\": None,\n",
    "    \"MDGAP-DEMENTIA\": None,\n",
    "    \"EXENATIDEPD3\": None,\n",
    "    \"GPDC\": None,\n",
    "    \"NORBANK\": None,\n",
    "    \"NYBB\": None,\n",
    "    \"NYPUM/PARKNY\": None,\n",
    "    \"UWBRAIN\": None,\n",
    "    \"SUREPD2\": None,\n",
    "    \"UNIBS\": None,\n",
    "    \"IUC\": None,\n",
    "    \"CANDAS-SMPD\": None,\n",
    "    \"CANDAS-PDGEN\": None,\n",
    "    \"EMORY\": None,\n",
    "    \"CANPARK\": None,\n",
    "    \"TAS\": None,\n",
    "    \"ISLAND\": None,\n",
    "    \"PPRBD\": None,\n",
    "    \"CRUCES\": None,\n",
    "    \"NRIBBB\": None,\n",
    "    \"NBB-HVD\": None,\n",
    "    \"NBB-SINAI\": None,\n",
    "    \"NBB-SPLVD\": None,\n",
    "    \"NBB-MLND\": None,\n",
    "    \"NBB-MIAMI\": None,\n",
    "    \"NBB-PBG\": None,\n",
    "    \"BELGRADE\": None,\n",
    "    \"MHHBB\": None,\n",
    "    \"SLOWSPEED\": None,\n",
    "    \"FMGC\": None,\n",
    "    \"IRAMS\": None,\n",
    "    \"GENOT\": None,\n",
    "    \"UTSW\": None,\n",
    "    \"ADPAR\": None,\n",
    "    \"NIMH\": None\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_map_df = pd.DataFrame(list(cohort_mapping.items()), columns=['gt', 'gmmb'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "tracker_merged = cohort_map_df.merge(gmmb_df, how='outer', left_on='gmmb', right_on='Cohort Name')\n",
    "tracker_merged.rename(columns={'gt':'genotracker'}, inplace=True)\n",
    "tracker_merged.drop(columns=['gmmb'], inplace=True)\n",
    "tracker_merged.to_csv('data/cohort_dashboard_updated.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## add data to database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "df_sub = df.loc[:,[\n",
    "    \"Study Code\",\n",
    "    \"Monogenic / Complex / Mixed\",\n",
    "    \"City\",\n",
    "    \"Geographic locality [free text]\",\n",
    "    \"DNA samples attempted [N]\",\n",
    "    \"Total samples passing basic QC [N]\",\n",
    "    \"Failed call rates [N < 95%]\",\n",
    "    \"Failed sex check [N with contradictory reported versus genetic sex]\",\n",
    "    \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\",\n",
    "    \"Duplicate samples [N, IBD estiamtes > 95%]\",\n",
    "    \"Imputation panels(s) [Free text]\",\n",
    "    \"Imputation complete for current samples [0 = no, 1 = yes]\",\n",
    "    \"Full data [0 = only summary stats, 1 = genotypes sharable]\",\n",
    "    \"GDPR [0 = no European restriction, 1 = limited by GDPR]\",\n",
    "    \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\"\n",
    "]]\n",
    "\n",
    "rename_dict = {\n",
    "    \"Study Code\": \"study_code\",\n",
    "    \"Monogenic / Complex / Mixed\": \"monogenic_complex_mixed\",\n",
    "    \"City\": \"city\",\n",
    "    \"Geographic locality [free text]\": \"geographic_locality\",\n",
    "    \"DNA samples attempted [N]\": \"n_dna_samples_attempted\",\n",
    "    \"Total samples passing basic QC [N]\": \"total_qc_pass\",\n",
    "    \"Failed call rates [N < 95%]\": \"callrate_fails\",\n",
    "    \"Failed sex check [N with contradictory reported versus genetic sex]\": \"sex_fails\",\n",
    "    \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\": \"het_fails\",\n",
    "    \"Duplicate samples [N, IBD estiamtes > 95%]\": \"duplicates\",\n",
    "    \"Imputation panels(s) [Free text]\": \"imputation_panel\",\n",
    "    \"Imputation complete for current samples [0 = no, 1 = yes]\": \"imputation_complete\",\n",
    "    \"Full data [0 = only summary stats, 1 = genotypes sharable]\": \"genotypes_shareable\",\n",
    "    \"GDPR [0 = no European restriction, 1 = limited by GDPR]\": \"gdpr\",\n",
    "    \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\": \"site\"\n",
    "}\n",
    "\n",
    "df_sub.rename(columns=rename_dict, inplace=True)\n",
    "\n",
    "df_out = df_sub.fillna({\n",
    "    'study_code': '',\n",
    "    'monogenic_complex_mixed': '',\n",
    "    'city': '',\n",
    "    'geographic_locality': '',\n",
    "    'n_dna_samples_attempted': 0,\n",
    "    'total_qc_pass': 0,\n",
    "    'callrate_fails': 0,\n",
    "    'sex_fails': 0,\n",
    "    'het_fails': 0,\n",
    "    'duplicates': 0,\n",
    "    'genotypes_shareable': 0,\n",
    "    'imputation_complete': 0,\n",
    "    'imputation_panel': '',\n",
    "    'gdpr': 0,\n",
    "    'site': ''\n",
    "})\n",
    "\n",
    "numeric_columns = [\n",
    "    'n_dna_samples_attempted', 'total_qc_pass', 'callrate_fails', 'sex_fails', \n",
    "    'het_fails', 'duplicates', 'genotypes_shareable', 'imputation_complete', 'gdpr'\n",
    "]\n",
    "\n",
    "df_out[numeric_columns] = df_out[numeric_columns].astype(int)\n",
    "\n",
    "ancestry_df = pd.read_csv('data/FEB_2024_cohort_ancestry_phenotype.csv')\n",
    "ancestry_df.rename(columns={'Cohort':'study_code'}, inplace=True)\n",
    "df_ancestry_out = df_out.merge(ancestry_df, on='study_code', how='left')\n",
    "columns_to_rename = list(ancestry_df.columns[1:])\n",
    "rename_mapping = {col: col.lower().replace(' ', '_') for col in columns_to_rename}\n",
    "df_ancestry_out.rename(columns=rename_mapping, inplace=True)\n",
    "df_ancestry_out.fillna(0, inplace=True)\n",
    "for col in rename_mapping.values():\n",
    "    df_ancestry_out[col] = df_ancestry_out[col].astype(int)\n",
    "\n",
    "df_ancestry_out.to_csv('data/genotracker_clean.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_code</th>\n",
       "      <th>monogenic_complex_mixed</th>\n",
       "      <th>city</th>\n",
       "      <th>geographic_locality</th>\n",
       "      <th>n_dna_samples_attempted</th>\n",
       "      <th>total_qc_pass</th>\n",
       "      <th>callrate_fails</th>\n",
       "      <th>sex_fails</th>\n",
       "      <th>het_fails</th>\n",
       "      <th>duplicates</th>\n",
       "      <th>...</th>\n",
       "      <th>cas_case</th>\n",
       "      <th>cas_control</th>\n",
       "      <th>cas_other</th>\n",
       "      <th>mde_case</th>\n",
       "      <th>mde_control</th>\n",
       "      <th>mde_other</th>\n",
       "      <th>cah_case</th>\n",
       "      <th>cah_control</th>\n",
       "      <th>cah_other</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BCM</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Houston,TX</td>\n",
       "      <td>USA</td>\n",
       "      <td>809</td>\n",
       "      <td>788</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CORIELL</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Camden, NJ</td>\n",
       "      <td>USA</td>\n",
       "      <td>8982</td>\n",
       "      <td>8718</td>\n",
       "      <td>140</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>40</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGNRTN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Miami, FL</td>\n",
       "      <td>USA</td>\n",
       "      <td>4694</td>\n",
       "      <td>3791</td>\n",
       "      <td>239</td>\n",
       "      <td>352</td>\n",
       "      <td>0</td>\n",
       "      <td>296</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UMD</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Baltimore, MD</td>\n",
       "      <td>USA</td>\n",
       "      <td>479</td>\n",
       "      <td>441</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PROSPOS</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Oslo</td>\n",
       "      <td>Norway</td>\n",
       "      <td>96</td>\n",
       "      <td>95</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>FMGC</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>IRAMS</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>GENOT</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Tunis</td>\n",
       "      <td>Tunisia</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>UTSW</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Dallas</td>\n",
       "      <td>United States</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>ADPAR</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bucharest</td>\n",
       "      <td>Romania</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    study_code monogenic_complex_mixed           city geographic_locality  \\\n",
       "0          BCM                 Complex     Houston,TX                 USA   \n",
       "1      CORIELL                 Complex     Camden, NJ                 USA   \n",
       "2      PDGNRTN                   Mixed     Miami, FL                  USA   \n",
       "3          UMD                 Complex  Baltimore, MD                 USA   \n",
       "4      PROSPOS                 Complex           Oslo              Norway   \n",
       "..         ...                     ...            ...                 ...   \n",
       "306       FMGC                   Mixed       Bordeaux              France   \n",
       "307      IRAMS                   Mixed       Bordeaux              France   \n",
       "308      GENOT                   Mixed          Tunis             Tunisia   \n",
       "309       UTSW                   Mixed         Dallas       United States   \n",
       "310      ADPAR                   Mixed      Bucharest             Romania   \n",
       "\n",
       "     n_dna_samples_attempted  total_qc_pass  callrate_fails  sex_fails  \\\n",
       "0                        809            788               7         12   \n",
       "1                       8982           8718             140         56   \n",
       "2                       4694           3791             239        352   \n",
       "3                        479            441               6         15   \n",
       "4                         96             95               0          1   \n",
       "..                       ...            ...             ...        ...   \n",
       "306                        0              0               0          0   \n",
       "307                        0              0               0          0   \n",
       "308                        0              0               0          0   \n",
       "309                        0              0               0          0   \n",
       "310                        0              0               0          0   \n",
       "\n",
       "     het_fails  duplicates  ... cas_case  cas_control  cas_other  mde_case  \\\n",
       "0            0           2  ...        0            0          0         0   \n",
       "1            1          40  ...        0            0          0         0   \n",
       "2            0         296  ...        6            0          0         4   \n",
       "3            0          17  ...        0            0          0         0   \n",
       "4            0           0  ...        0            0          0         0   \n",
       "..         ...         ...  ...      ...          ...        ...       ...   \n",
       "306          0           0  ...        0            0          0         0   \n",
       "307          0           0  ...        0            0          0         0   \n",
       "308          0           0  ...        0            0          0         0   \n",
       "309          0           0  ...        0            0          0         0   \n",
       "310          0           0  ...        0            0          0         0   \n",
       "\n",
       "    mde_control  mde_other  cah_case  cah_control  cah_other  total  \n",
       "0             0          0         0            0          0      0  \n",
       "1             0          0         0            0          0      0  \n",
       "2             0          0        29            0          0   2972  \n",
       "3             0          0         0            0          0      0  \n",
       "4             0          0         0            0          0      0  \n",
       "..          ...        ...       ...          ...        ...    ...  \n",
       "306           0          0         0            0          0      0  \n",
       "307           0          0         0            0          0      0  \n",
       "308           0          0         0            0          0      0  \n",
       "309           0          0         0            0          0      0  \n",
       "310           0          0         0            0          0      0  \n",
       "\n",
       "[311 rows x 49 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ancestry_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "!python3 load_data.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### OLD genotracker processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "# df_sub = df.loc[:,[\n",
    "#     \"Study Code\",\n",
    "#     \"Monogenic / Complex / Mixed\",\n",
    "#     \"City\",\n",
    "#     \"Geographic locality [free text]\",\n",
    "#     \"DNA samples attempted [N]\",\n",
    "#     \"Total samples passing basic QC [N]\",\n",
    "#     \"Failed call rates [N < 95%]\",\n",
    "#     \"Failed sex check [N with contradictory reported versus genetic sex]\",\n",
    "#     \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\",\n",
    "#     \"Duplicate samples [N, IBD estiamtes > 95%]\",\n",
    "#     \"African (AFR)\",\n",
    "#     \"African admixed (AAC)\",\n",
    "#     \"Ashkenzai (AJ)\",\n",
    "#     \"East Asian (EAS)\",\n",
    "#     \"European (EUR)\",\n",
    "#     \"Finnish (FIN)\",\n",
    "#     \"LatinX (AMR)\",\n",
    "#     \"South Asian (SAS)\",\n",
    "#     \"Central Asian (CAS)\",\n",
    "#     \"Middle Eastern (MDE)\",\n",
    "#     \"Complex Admixture (CAH)\",\n",
    "#     \"Total\",\n",
    "#     \"Imputation panels(s) [Free text]\",\n",
    "#     \"Imputation complete for current samples [0 = no, 1 = yes]\",\n",
    "#     \"Full data [0 = only summary stats, 1 = genotypes sharable]\",\n",
    "#     \"GDPR [0 = no European restriction, 1 = limited by GDPR]\",\n",
    "#     \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\"\n",
    "# ]]\n",
    "\n",
    "# rename_dict = {\n",
    "#     \"Study Code\": \"study_code\",\n",
    "#     \"Monogenic / Complex / Mixed\": \"monogenic_complex_mixed\",\n",
    "#     \"City\": \"city\",\n",
    "#     \"Geographic locality [free text]\": \"geographic_locality\",\n",
    "#     \"DNA samples attempted [N]\": \"n_dna_samples_attempted\",\n",
    "#     \"Total samples passing basic QC [N]\": \"total_qc_pass\",\n",
    "#     \"Failed call rates [N < 95%]\": \"callrate_fails\",\n",
    "#     \"Failed sex check [N with contradictory reported versus genetic sex]\": \"sex_fails\",\n",
    "#     \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\": \"het_fails\",\n",
    "#     \"Duplicate samples [N, IBD estiamtes > 95%]\": \"duplicates\",\n",
    "#     \"African (AFR)\": \"AFR\",\n",
    "#     \"African admixed (AAC)\": \"AAC\",\n",
    "#     \"Ashkenzai (AJ)\": \"AJ\",\n",
    "#     \"East Asian (EAS)\": \"EAS\",\n",
    "#     \"European (EUR)\": \"EUR\",\n",
    "#     \"Finnish (FIN)\": \"FIN\",\n",
    "#     \"LatinX (AMR)\": \"AMR\",\n",
    "#     \"South Asian (SAS)\": \"SAS\",\n",
    "#     \"Central Asian (CAS)\": \"CAS\",\n",
    "#     \"Middle Eastern (MDE)\": \"MDE\",\n",
    "#     \"Complex Admixture (CAH)\": \"CAH\",\n",
    "#     \"Total\": \"total\",\n",
    "#     \"Imputation panels(s) [Free text]\": \"imputation_panel\",\n",
    "#     \"Imputation complete for current samples [0 = no, 1 = yes]\": \"imputation_complete\",\n",
    "#     \"Full data [0 = only summary stats, 1 = genotypes sharable]\": \"genotypes_shareable\",\n",
    "#     \"GDPR [0 = no European restriction, 1 = limited by GDPR]\": \"gdpr\",\n",
    "#     \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\": \"site\"\n",
    "# }\n",
    "\n",
    "# df_sub.rename(columns=rename_dict, inplace=True)\n",
    "\n",
    "# df_out = df_sub.fillna({\n",
    "#     'study_code': '',\n",
    "#     'monogenic_complex_mixed': '',\n",
    "#     'city': '',\n",
    "#     'geographic_locality': '',\n",
    "#     'n_dna_samples_attempted': 0,\n",
    "#     'total_qc_pass': 0,\n",
    "#     'callrate_fails': 0,\n",
    "#     'sex_fails': 0,\n",
    "#     'het_fails': 0,\n",
    "#     'duplicates': 0,\n",
    "#     'AFR': 0,\n",
    "#     'AAC': 0,\n",
    "#     'AJ': 0,\n",
    "#     'EAS': 0,\n",
    "#     'EUR': 0,\n",
    "#     'FIN': 0,\n",
    "#     'AMR': 0,\n",
    "#     'SAS': 0,\n",
    "#     'CAS': 0,\n",
    "#     'MDE': 0,\n",
    "#     'CAH': 0,\n",
    "#     'total': 0,\n",
    "#     'genotypes_shareable': 0,\n",
    "#     'imputation_complete': 0,\n",
    "#     'imputation_panel': '',\n",
    "#     'gdpr': 0,\n",
    "#     'site': ''\n",
    "# })\n",
    "\n",
    "# numeric_columns = [\n",
    "#     'n_dna_samples_attempted', 'total_qc_pass', 'callrate_fails', 'sex_fails', \n",
    "#     'het_fails', 'duplicates', 'AFR', 'AAC', 'AJ', 'EAS', 'EUR', 'FIN', \n",
    "#     'AMR', 'SAS', 'CAS', 'MDE', 'CAH', 'total', 'genotypes_shareable', \n",
    "#     'imputation_complete', 'gdpr'\n",
    "# ]\n",
    "\n",
    "# df_out[numeric_columns] = df_out[numeric_columns].astype(int)\n",
    "# df_out.to_csv('data/genotracker_clean.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## use database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "engine = create_engine('sqlite:///test.db')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "from crud import update_fields_by_study_code\n",
    "from datetime import date\n",
    "\n",
    "# Example update\n",
    "updates = {\n",
    "    \"city\": \"BLAH\",\n",
    "    \"total_qc_pass\": 95,\n",
    "    \"date_last_update\": date.today(),\n",
    "    \"new\": False\n",
    "}\n",
    "update_fields_by_study_code(\"BCM\", updates)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   id study_code monogenic_complex_mixed           city geographic_locality  \\\n",
      "0   1        BCM                 Complex           BLAH                 USA   \n",
      "1   2    CORIELL                 Complex     Camden, NJ                 USA   \n",
      "2   3    PDGNRTN                   Mixed     Miami, FL                  USA   \n",
      "3   4        UMD                 Complex  Baltimore, MD                 USA   \n",
      "4   5    PROSPOS                 Complex           Oslo              Norway   \n",
      "\n",
      "   n_dna_samples_attempted  total_qc_pass  callrate_fails  sex_fails  \\\n",
      "0                      809             95               7         12   \n",
      "1                     8982           8718             140         56   \n",
      "2                     4694           3791             239        352   \n",
      "3                      479            441               6         15   \n",
      "4                       96             95               0          1   \n",
      "\n",
      "   het_fails  ...  mde_case  mde_control mde_other  cah_case  cah_control  \\\n",
      "0          0  ...         0            0         0         0            0   \n",
      "1          1  ...         0            0         0         0            0   \n",
      "2          0  ...         4            0         0        29            0   \n",
      "3          0  ...         0            0         0         0            0   \n",
      "4          0  ...         0            0         0         0            0   \n",
      "\n",
      "   cah_other total  new  date_last_update  compliance  \n",
      "0          0     0    0        2024-06-28           0  \n",
      "1          0     0    1        2024-06-28           0  \n",
      "2          0  2972    1        2024-06-28           0  \n",
      "3          0     0    1        2024-06-28           0  \n",
      "4          0     0    1        2024-06-28           0  \n",
      "\n",
      "[5 rows x 54 columns]\n"
     ]
    }
   ],
   "source": [
    "query = \"SELECT * FROM cohort_data\"\n",
    "df = pd.read_sql(query, engine)\n",
    "\n",
    "print(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>study_code</th>\n",
       "      <th>monogenic_complex_mixed</th>\n",
       "      <th>city</th>\n",
       "      <th>geographic_locality</th>\n",
       "      <th>n_dna_samples_attempted</th>\n",
       "      <th>total_qc_pass</th>\n",
       "      <th>callrate_fails</th>\n",
       "      <th>sex_fails</th>\n",
       "      <th>het_fails</th>\n",
       "      <th>...</th>\n",
       "      <th>mde_case</th>\n",
       "      <th>mde_control</th>\n",
       "      <th>mde_other</th>\n",
       "      <th>cah_case</th>\n",
       "      <th>cah_control</th>\n",
       "      <th>cah_other</th>\n",
       "      <th>total</th>\n",
       "      <th>new</th>\n",
       "      <th>date_last_update</th>\n",
       "      <th>compliance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>BCM</td>\n",
       "      <td>Complex</td>\n",
       "      <td>NewCityName</td>\n",
       "      <td>USA</td>\n",
       "      <td>809</td>\n",
       "      <td>95</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Camden, NJ</td>\n",
       "      <td>USA</td>\n",
       "      <td>8982</td>\n",
       "      <td>8718</td>\n",
       "      <td>140</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>PDGNRTN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Miami, FL</td>\n",
       "      <td>USA</td>\n",
       "      <td>4694</td>\n",
       "      <td>3791</td>\n",
       "      <td>239</td>\n",
       "      <td>352</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2972</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>UMD</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Baltimore, MD</td>\n",
       "      <td>USA</td>\n",
       "      <td>479</td>\n",
       "      <td>441</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>PROSPOS</td>\n",
       "      <td>Complex</td>\n",
       "      <td>Oslo</td>\n",
       "      <td>Norway</td>\n",
       "      <td>96</td>\n",
       "      <td>95</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>307</td>\n",
       "      <td>FMGC</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>308</td>\n",
       "      <td>IRAMS</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bordeaux</td>\n",
       "      <td>France</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>309</td>\n",
       "      <td>GENOT</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Tunis</td>\n",
       "      <td>Tunisia</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>310</td>\n",
       "      <td>UTSW</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Dallas</td>\n",
       "      <td>United States</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>311</td>\n",
       "      <td>ADPAR</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>Bucharest</td>\n",
       "      <td>Romania</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-06-28</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id study_code monogenic_complex_mixed           city  \\\n",
       "0      1        BCM                 Complex    NewCityName   \n",
       "1      2    CORIELL                 Complex     Camden, NJ   \n",
       "2      3    PDGNRTN                   Mixed     Miami, FL    \n",
       "3      4        UMD                 Complex  Baltimore, MD   \n",
       "4      5    PROSPOS                 Complex           Oslo   \n",
       "..   ...        ...                     ...            ...   \n",
       "306  307       FMGC                   Mixed       Bordeaux   \n",
       "307  308      IRAMS                   Mixed       Bordeaux   \n",
       "308  309      GENOT                   Mixed          Tunis   \n",
       "309  310       UTSW                   Mixed         Dallas   \n",
       "310  311      ADPAR                   Mixed      Bucharest   \n",
       "\n",
       "    geographic_locality  n_dna_samples_attempted  total_qc_pass  \\\n",
       "0                   USA                      809             95   \n",
       "1                   USA                     8982           8718   \n",
       "2                   USA                     4694           3791   \n",
       "3                   USA                      479            441   \n",
       "4                Norway                       96             95   \n",
       "..                  ...                      ...            ...   \n",
       "306              France                        0              0   \n",
       "307              France                        0              0   \n",
       "308             Tunisia                        0              0   \n",
       "309       United States                        0              0   \n",
       "310             Romania                        0              0   \n",
       "\n",
       "     callrate_fails  sex_fails  het_fails  ...  mde_case  mde_control  \\\n",
       "0                 7         12          0  ...         0            0   \n",
       "1               140         56          1  ...         0            0   \n",
       "2               239        352          0  ...         4            0   \n",
       "3                 6         15          0  ...         0            0   \n",
       "4                 0          1          0  ...         0            0   \n",
       "..              ...        ...        ...  ...       ...          ...   \n",
       "306               0          0          0  ...         0            0   \n",
       "307               0          0          0  ...         0            0   \n",
       "308               0          0          0  ...         0            0   \n",
       "309               0          0          0  ...         0            0   \n",
       "310               0          0          0  ...         0            0   \n",
       "\n",
       "    mde_other  cah_case  cah_control  cah_other total  new  date_last_update  \\\n",
       "0           0         0            0          0     0    0        2024-06-28   \n",
       "1           0         0            0          0     0    1        2024-06-28   \n",
       "2           0        29            0          0  2972    1        2024-06-28   \n",
       "3           0         0            0          0     0    1        2024-06-28   \n",
       "4           0         0            0          0     0    1        2024-06-28   \n",
       "..        ...       ...          ...        ...   ...  ...               ...   \n",
       "306         0         0            0          0     0    1        2024-06-28   \n",
       "307         0         0            0          0     0    1        2024-06-28   \n",
       "308         0         0            0          0     0    1        2024-06-28   \n",
       "309         0         0            0          0     0    1        2024-06-28   \n",
       "310         0         0            0          0     0    1        2024-06-28   \n",
       "\n",
       "     compliance  \n",
       "0             0  \n",
       "1             0  \n",
       "2             0  \n",
       "3             0  \n",
       "4             0  \n",
       "..          ...  \n",
       "306           0  \n",
       "307           0  \n",
       "308           0  \n",
       "309           0  \n",
       "310           0  \n",
       "\n",
       "[311 rows x 54 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
