{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import difflib\n",
    "from sqlalchemy import create_engine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gt_df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "gmmb_df = pd.read_csv('data/gmmb_tracker.csv', header=1)\n",
    "ancestry_df = pd.read_csv('data/FEB_2024_cohort_ancestry_phenotype.csv', header=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## match gp2 'Study Code' to gmmb 'Cohort Name'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# def find_best_matches(list1, list2, cutoff=0.6):\n",
    "#     matches = {}\n",
    "#     for item in list1:\n",
    "#         match = difflib.get_close_matches(item, list2, n=1, cutoff=cutoff)\n",
    "#         if match:\n",
    "#             matches[item] = match[0]\n",
    "#         else:\n",
    "#             matches[item] = None\n",
    "#     return matches\n",
    "\n",
    "# best_matches = find_best_matches(gt_df.loc[~gt_df['Study Code'].isna(), 'Study Code'], gmmb_df['Cohort Name'])\n",
    "\n",
    "# for k, v in best_matches.items():\n",
    "#     print(f\"{k}: {v}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 715,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genotracker\n",
      "Study Code: NIMH\n",
      "Individual Study Name: The genetics of REM sleep behavior disorder, parkinson´s disease and other synucleinopathies\n",
      "Institution: National Institute of Mental Health\n",
      "City: Prague\n",
      "Geographic locality [free text]: Czech Republic\n",
      "\n",
      "############################################################\n",
      "\n",
      "gmmb\n",
      "Cohort Name: LUX GIANT - National Institute of Mental Health and Neurosciences\n",
      "Institution: National Institute of Mental Health and Neurosciences\n",
      "City: Bangalore\n",
      "Country (Location): India\n",
      "\n",
      "############################################################\n",
      "\n"
     ]
    }
   ],
   "source": [
    "study_code = \"ADPAR\"\n",
    "\n",
    "gt_fields = ['Study Code', 'Individual Study Name', 'Institution', 'City', 'Geographic locality [free text]']\n",
    "# gt_fields_df = gt_df.loc[gt_df['Study Code']==study_code, gt_fields]\n",
    "gt_fields_df = gt_df.loc[gt_df['Institution'].str.contains(\"National Institute of Mental Health\", na=False), gt_fields]\n",
    "\n",
    "gmmb_fields = ['Cohort Name', 'Institution', 'City', 'Country (Location)']\n",
    "gmmb_fields_df = gmmb_df.loc[gmmb_df[\"Cohort Name\"].str.contains(\"National Institute of Mental Health and Neurosciences\"), gmmb_fields]\n",
    "\n",
    "\n",
    "print('genotracker')\n",
    "for index, row in gt_fields_df.iterrows():\n",
    "    for col_name, value in row.items():\n",
    "        print(f'{col_name}: {value}')\n",
    "    print()\n",
    "\n",
    "print('##'*30)\n",
    "print()\n",
    "\n",
    "print('gmmb')\n",
    "for index, row in gmmb_fields_df.iterrows():\n",
    "    for col_name, value in row.items():\n",
    "        print(f'{col_name}: {value}')\n",
    "    print()\n",
    "    \n",
    "print('##'*30)\n",
    "print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 726,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "167"
      ]
     },
     "execution_count": 726,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# gmmb_df['Name on Ancestry Phenotype Backup']\n",
    "# gt_df.merge(gmmb_df, how='left', left_on='Study Code', right_on='Name on Ancestry Phenotype Backup')\n",
    "matching_cohorts = {\n",
    "    \"BCM\": \"Baylor College of Medicine (BCM)\",\n",
    "    \"CORIELL\": \"CORIELL\",\n",
    "    \"PDGNRTN\": \"PDGENEration\",\n",
    "    \"UMD\": \"BCM-UMD\",\n",
    "    \"PROSPOS\": \"PROSPOS\",\n",
    "    \"SP-SEV\": \"Sp_SEV - HUVR/CSIC/Universidad de Sevill\",\n",
    "    \"COPN\": \"C-OPN\",\n",
    "    \"NZP3\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"UPENNU19\": \"UPENN U19 Cohort\",\n",
    "    \"BADGEPD\": \"BADGE-PD\",\n",
    "    \"CAMPAIGN\": \"CamPaIGN\",\n",
    "    \"DIGPD\": \"DIGPD\",\n",
    "    \"ICEBERG\": \"ICEBERG\",\n",
    "    \"IDCME\": \"NGC\",\n",
    "    \"MIND\": \"MIND Initiative\",\n",
    "    \"NSPARK\": \"NS PARK\",\n",
    "    \"PICNICS\": \"PICNICS\",\n",
    "    \"PREDISTIM\": \"PREDICT-PD\",\n",
    "    \"QPN\": \"Quebec Parkinson's Network\",\n",
    "    \"RBM\": \"RBM\",\n",
    "    \"SP-BCN3\": \"Sp_BCN3 - Sant Joan de Déu\",\n",
    "    \"SP-OVI\": \"Sp_OVI - University of Oviedo\",\n",
    "    \"MDGAP-KINGS\": \"MD GAP - King's College London\",\n",
    "    \"MDGAP-QSBB\": \"MD GAP - QSBB - University College London\",\n",
    "    \"POM\": \"POM\",\n",
    "    \"LCC\": \"LCC\",\n",
    "    \"UMKLM\": \"Malaysian Parkinson's Genetics Cohort - UMKLM\",\n",
    "    \"GPARKGR\": \"GenePARKGR\",\n",
    "    \"PROPARK\": \"ProPark\",\n",
    "    \"SP-BCN2\": \"Sp_BCN2 - University Hospital Mútua de Terrassa\",\n",
    "    \"SP-SNT\": \"Sp_SNT - University Hospital Marques de Valdecilla-IDIVAL\",\n",
    "    \"BIOFIND\": \"BioFIND\",\n",
    "    \"BLAACPD-KPM\": \"BLAACPD - Kaiser Permanente\",\n",
    "    \"BLAACPD-RUSH\": \"BLAACPD - Rush University\",\n",
    "    \"BLAACPD-UAB\": \"BLAACPD - University of Alabama at Birmingham\",\n",
    "    \"BLAACPD-UC\": \"BLAACPD - University of Chicago\",\n",
    "    \"CALYPSO\": \"CALYPSO\",\n",
    "    \"MDGAP-EBB\": \"MD GAP - Edinburgh Brain Bank - University of Edinburgh\",\n",
    "    \"PFP\": \"PFP\",\n",
    "    \"PPMI\": \"PPMI\",\n",
    "    \"PROBAND\": \"PROBAND\",\n",
    "    \"SYNAPS-KZ\": \"SYNAPS-KZ\",\n",
    "    \"VIPD\": \"VIPD\",\n",
    "    \"CANDAS\": \"CANDAS\",\n",
    "    \"QPP\": \"QPP\",\n",
    "    \"UGUPD\": \"UGCPD\",\n",
    "    \"GENEFINDER\": \"GeneFinder\",\n",
    "    \"OPDC\": \"OPDC Discovery Cohort\",\n",
    "    \"MDGAP-MBB\": \"Manchester Brain Bank\",\n",
    "    \"PROMO\": \"ProtectMove - Institute of Neurogenetics, University of Lübeck\",\n",
    "    \"PROATYP\": \"proATYP\",\n",
    "    \"MDGAP-IBB\": \"PUKBB\",\n",
    "    \"BENHAMIDA\": \"Clinicogenetic Study of PD in Tunisia\",\n",
    "    \"ELPD\": \"ELPD Project\",\n",
    "    \"PDSGR\": \"PDS\",\n",
    "    \"PREDICTPD\": \"PREDICT-PD\",\n",
    "    \"AUTH\": \"AUTH - GP2 study\",\n",
    "    \"PDSTAT\": \"PD STAT\",\n",
    "    \"MPBC\": \"MPBC\",\n",
    "    \"SYDPDRC\": \"Sydney PDRC\",\n",
    "    \"APGS\": \"APGS\",\n",
    "    \"SYDBB\": \"Sydney Brain Bank - SYDBB\",\n",
    "    \"CPRC\": \"Cambridge Parkinson's Research Clinic\",\n",
    "    \"IPDGCEA-JKU\": \"IPDGC East Asia - Jikei University\",\n",
    "    \"ADEPTPD\": \"ADepT-PD\",\n",
    "    \"PDMDC\": \"PDMDC/NUgene\",\n",
    "    \"GARVAN\": \"Genetic analysis in inherited neurological disorders\",\n",
    "    \"KUL\": \"Genotype-Phenotype Correlations in PD and Related Movement Disorders - KUL\",\n",
    "    \"PDMDCB\": \"Parkinson Disease and Movement Disorders Center Biorepository\",\n",
    "    \"TUEPAC\": \"Tuepac + TREND\",\n",
    "    \"LUXPARK\": \"LUX-Park\",\n",
    "    \"BIOPARK\": \"To be updated - Karolinska Institutet\",\n",
    "    \"IMMLISBON\": \"Movement Disorders Biobanc\",\n",
    "    \"PDGEN\": \"PDGENEration\",\n",
    "    \"DATATOP\": \"DATATOP\",\n",
    "    \"SZTE\": \"SZTE\",\n",
    "    \"TRD\": \"Trondheim Study\",\n",
    "    \"MDGAP-SW\": \"MD GAP - South West Dementia Brain Bank - Southmead Hospital\",\n",
    "    \"S4\": \"S4\",\n",
    "    \"NETPDLS1\": \"NET-PD\",\n",
    "    \"DENOPA\": \"DeNoPa\",\n",
    "    \"NTUH\": \"Genomewide Association Study in PD among East Asians - NTUH\",\n",
    "    \"BRFAA\": \"Biological markers in Parkinson's disease - BRFAA\",\n",
    "    \"BUE\": \"BUE - GP2\",\n",
    "    \"LARGEPD\": \"To be updated - Cleveland Clinic\",\n",
    "    \"BBDP\": \"Arizona Brain Bank Brain and body donation programme (BBDP)\",\n",
    "    \"CINCINNATI\": \"Clinical and genetic study of Unusual/Familial Movement Disorders\",\n",
    "    \"MMCTU\": \"Genotype-Phenotype Correlations in Nepalese PD Population - MMCTU\",\n",
    "    \"PERUGIA\": \"CSF Biomarkers\",\n",
    "    \"STELLENBOS\": \"Genetic study of PD and related movement disorders\",\n",
    "    \"CHULALONGK\": \"Genetic Testing in Patients with Movement Disorders and PD\",\n",
    "    \"MDGAP-NWCSTL\": \"MD GAP - Newcastle Brain Bank - University of Newcastle\",\n",
    "    \"KFSH\": \"The Molecular Basis of Movement Disorders in Saudis - KFSH\",\n",
    "    \"PAGE\": \"PAGE\",\n",
    "    \"SYNAPS-GR\": \"SYNAPS-GR\",\n",
    "    \"PDCARE\": \"PDCARE\",\n",
    "    \"RAPSODI\": \"RAPSODI\",\n",
    "    \"YOPDFL\": \"YOPDFL (Young Onset Parkinson's Disease)\",\n",
    "    \"FDSANTIAGO\": \"GP2 - Fundación Diagnosis\",\n",
    "    \"MULODZ\": \"Molecular Testing of Polish PD Patients\",\n",
    "    \"TONIC-PERRON\": \"TONiC PD\",\n",
    "    \"KOC\": \"GP2 - Koc University\",\n",
    "    \"JGUM\": \"JGUM\",\n",
    "    \"PATRAS\": \"PATRAS\",\n",
    "    \"NUGENE\": \"NU Gene\",\n",
    "    \"CDPDGR\": \"Thessaly PD Database\",\n",
    "    \"JENPARK\": \"Jena Parkinson Study\",\n",
    "    \"UROMETV\": \"Standardized data collection of PD patients\",\n",
    "    \"CHCHPD\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"HIPDR\": \"HIPDR\",\n",
    "    \"KOLLING\": \"Kolling PD\",\n",
    "    \"SEOULNUH\": \"GP2 - Seoul National University Hospital\",\n",
    "    \"NROSF2U\": \"Multicentric Retrospective and Prospective PD Database\",\n",
    "    \"BLAACPD-LSU\": \"BLAACPD - Louisiana State University\",\n",
    "    \"IMMUNEPD\": \"Innate and Adaptive Immunity in PD\",\n",
    "    \"PROMO-EPIPARK\": \"PROMO-EPIPARK\",\n",
    "    \"PROMO-GENEPARK\": \"PROMO-GENEPARK\",\n",
    "    \"PROMO-ROPAD\": \"PROMO-ROPAD\",\n",
    "    \"PROMO-SFB134\": \"PROMO-SFB134\",\n",
    "    \"PROMO-SYSMED\": \"PROMO-SYSMEDPD\",\n",
    "    \"OXQUIP\": \"Oxford Study of Quantification in Parkinsonism\",\n",
    "    \"PROMO-USTP\": \"PROMO-USTP\",\n",
    "    \"MUVIENNA\": \"Families with Neurodegenerative Diseases\",\n",
    "    \"CATANZARO\": \"Familial and Early-onset PD Database\",\n",
    "    \"QMAP\": \"Quantitative MRI for Anatomical Phenotyping in Parkinson's Disease\",\n",
    "    \"CATPD\": \"Central Asian and Transcaucasian PD (CAT-PD)\",\n",
    "    \"IPDGCAF-NG\": \"IPDGC Africa - The Nigerian Parkinson Disease Research Network\",\n",
    "    \"IPDGCEA-JU\": \"IPDGC East Asia - Juntendo University\",\n",
    "    \"TUMBB\": \"Biobank for Diseases of the Peripheral and Central Nervous System - TUMBB\",\n",
    "    \"TUMPD\": \"Genetics of Parkinson's Disease - TUMPD\",\n",
    "    \"ANANEOS\": \"ANANEOS Study\",\n",
    "    \"MDGAP-CAMBRIDGE\": \"MD GAP - Cambridge\",\n",
    "    \"MDGAP-OXF\": \"MD GAP - OXF\",\n",
    "    \"VETPD\": \"VET-PD\",\n",
    "    \"BERNGEN\": 'BERN',\n",
    "    \"SINAI\": 'Mount Sinai Cohort',\n",
    "    \"PARKWEST\": \"The Norwegian Parkwest Study\",\n",
    "    \"PROSPECT\": \"PROSPECT\",\n",
    "    \"UMUDALL\": \"UMUDALL\",\n",
    "    \"TORONTO\": \"To be updated - University of Toronto\",\n",
    "    \"MDGAP-VBB\": \"MD GAP - Victorian Brain Bank\",\n",
    "    \"IRCCSN\": \"Genetics of Neurological Pathologies: New Variations, New Phenotypes\",\n",
    "    \"PROBEPD\": \"PROBEPD\",\n",
    "    \"TLVMC\": \"DLB Cohort TLVMC\",\n",
    "    \"ANDPD\": \"ANDPD\",    \n",
    "    \"MONPAR\": \"Genetic Study for Pakinson's Disease in Mongolia\",\n",
    "    \"NGIPDO\": \"NGIPDO - Neurogenomics and Informatics Center - Parkinson's Disease  Deep Molecular Profiling\",\n",
    "    \"IPDGCEA-UKM\": \"IPDGC East Asia - UKM\",\n",
    "    \"DODONA\": \"The DodoNa Project: DNA Predictions to Improve Neurological Health\",\n",
    "    \"PROMO-CARRIERS\": \"PROMO-CARRIERS\",\n",
    "    \"HAWAIIPD\": \"HAWAII PD\",\n",
    "    \"YMS\": \"Yale Medical School\",\n",
    "    \"IGECND\": \"IGECND\",\n",
    "    \"DHAKA\": \"GP2 - Bangladesh\",\n",
    "    \"UGCPD\": \"UGCPD\",\n",
    "    \"AAPDGC\": \"AfrAbia-PD Genomic Consortium (AA-PDGC)\",\n",
    "    \"REGARDS\": \"REasons for Geographic and Racial Differences in Stroke (REGARDS) Project\",\n",
    "    \"NZPEGS\": \"New Zealand Parkinson's Environment and Genes Study\",\n",
    "    \"GPX\": \"GenePool Study\",\n",
    "    \"VAIBB\": \"Van Andel Institute Brain Bank\",\n",
    "    \"WEHI\": \"WEHI - Integrating molecular and omics approaches in Parkinson’s disease\",\n",
    "    \"ACCESSPD\": \"ACCESSPD\",\n",
    "    \"GPS\": \"GenePool Study\",\n",
    "    \"NGP\": \"The Neuro Genomics Partnership\",\n",
    "    \"SCANPSP\": \"SCANPSP - Study of Comprehensive ANd Multimodal Marker-based Cohort of PSP\",\n",
    "    \"SAMSUNG\": \"GP2 - Samsung\",\n",
    "    \"BURGOS\": \"Local Study - Burgos\"\n",
    "}\n",
    "\n",
    "len(matching_cohorts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 725,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gt_name_merge = gt_df[['Study Code','Individual Study Name']].merge(gmmb_df[['Cohort Name']], left_on='Individual Study Name', right_on='Cohort Name')\n",
    "# for index, row in gt_name_merge.iterrows():\n",
    "#     print(f'Row {index}')\n",
    "#     for col_name, value in row.items():\n",
    "#         print(f\" {col_name}: {value}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_mapping = {\n",
    "    \"BCM\": \"Baylor College of Medicine (BCM)\",\n",
    "    \"CORIELL\": \"CORIELL\",\n",
    "    \"PDGNRTN\": \"PDGENEration\",\n",
    "    \"UMD\": \"BCM-UMD\",\n",
    "    \"PROSPOS\": \"PROSPOS\",\n",
    "    \"SP-SEV\": \"Sp_SEV - HUVR/CSIC/Universidad de Sevill\",\n",
    "    \"COPN\": \"C-OPN\",\n",
    "    \"NZP3\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"UPENNU19\": \"UPENN U19 Cohort\",\n",
    "    \"BADGEPD\": \"BADGE-PD\",\n",
    "    \"CAMPAIGN\": \"CamPaIGN\",\n",
    "    \"DIGPD\": \"DIGPD\",\n",
    "    \"ICEBERG\": \"ICEBERG\",\n",
    "    \"IDCME\": \"NGC\",\n",
    "    \"MIND\": \"MIND Initiative\",\n",
    "    \"NSPARK\": \"NS PARK\",\n",
    "    \"PICNICS\": \"PICNICS\",\n",
    "    \"PREDISTIM\": \"PREDICT-PD\",\n",
    "    \"QPN\": \"Quebec Parkinson's Network\",\n",
    "    \"RBM\": \"RBM\",\n",
    "    \"SP-BCN3\": \"Sp_BCN3 - Sant Joan de Déu\",\n",
    "    \"SP-OVI\": \"Sp_OVI - University of Oviedo\",\n",
    "    \"MDGAP-KINGS\": \"MD GAP - King's College London\",\n",
    "    \"MDGAP-QSBB\": \"MD GAP - QSBB - University College London\",\n",
    "    \"POM\": \"POM\",\n",
    "    \"LCC\": \"LCC\",\n",
    "    \"UMKLM\": \"Malaysian Parkinson's Genetics Cohort - UMKLM\",\n",
    "    \"GPARKGR\": \"GenePARKGR\",\n",
    "    \"PROPARK\": \"ProPark\",\n",
    "    \"SP-BCN2\": \"Sp_BCN2 - University Hospital Mútua de Terrassa\",\n",
    "    \"SP-SNT\": \"Sp_SNT - University Hospital Marques de Valdecilla-IDIVAL\",\n",
    "    \"BIOFIND\": \"BioFIND\",\n",
    "    \"BLAACPD-KPM\": \"BLAACPD - Kaiser Permanente\",\n",
    "    \"BLAACPD-RUSH\": \"BLAACPD - Rush University\",\n",
    "    \"BLAACPD-UAB\": \"BLAACPD - University of Alabama at Birmingham\",\n",
    "    \"BLAACPD-UC\": \"BLAACPD - University of Chicago\",\n",
    "    \"CALYPSO\": \"CALYPSO\",\n",
    "    \"MDGAP-EBB\": \"MD GAP - Edinburgh Brain Bank - University of Edinburgh\",\n",
    "    \"PFP\": \"PFP\",\n",
    "    \"PPMI\": \"PPMI\",\n",
    "    \"PROBAND\": \"PROBAND\",\n",
    "    \"SYNAPS-KZ\": \"SYNAPS-KZ\",\n",
    "    \"VIPD\": \"VIPD\",\n",
    "    \"CANDAS\": \"CANDAS\",\n",
    "    \"QPP\": \"QPP\",\n",
    "    \"UGUPD\": \"UGCPD\",\n",
    "    \"GENEFINDER\": \"GeneFinder\",\n",
    "    \"OPDC\": \"OPDC Discovery Cohort\",\n",
    "    \"MDGAP-MBB\": \"Manchester Brain Bank\",\n",
    "    \"PROMO\": \"ProtectMove - Institute of Neurogenetics, University of Lübeck\",\n",
    "    \"PROATYP\": \"proATYP\",\n",
    "    \"MDGAP-IBB\": \"PUKBB\",\n",
    "    \"BENHAMIDA\": \"Clinicogenetic Study of PD in Tunisia\",\n",
    "    \"ELPD\": \"ELPD Project\",\n",
    "    \"PDSGR\": \"PDS\",\n",
    "    \"PREDICTPD\": \"PREDICT-PD\",\n",
    "    \"AUTH\": \"AUTH - GP2 study\",\n",
    "    \"PDSTAT\": \"PD STAT\",\n",
    "    \"MPBC\": \"MPBC\",\n",
    "    \"SYDPDRC\": \"Sydney PDRC\",\n",
    "    \"APGS\": \"APGS\",\n",
    "    \"SYDBB\": \"Sydney Brain Bank - SYDBB\",\n",
    "    \"CPRC\": \"Cambridge Parkinson's Research Clinic\",\n",
    "    \"IPDGCEA-JKU\": \"IPDGC East Asia - Jikei University\",\n",
    "    \"ADEPTPD\": \"ADepT-PD\",\n",
    "    \"PDMDC\": \"PDMDC/NUgene\",\n",
    "    \"GARVAN\": \"Genetic analysis in inherited neurological disorders\",\n",
    "    \"KUL\": \"Genotype-Phenotype Correlations in PD and Related Movement Disorders - KUL\",\n",
    "    \"PDMDCB\": \"Parkinson Disease and Movement Disorders Center Biorepository\",\n",
    "    \"TUEPAC\": \"Tuepac + TREND\",\n",
    "    \"LUXPARK\": \"LUX-Park\",\n",
    "    \"BIOPARK\": \"To be updated - Karolinska Institutet\",\n",
    "    \"IMMLISBON\": \"Movement Disorders Biobanc\",\n",
    "    \"PDGEN\": \"PDGENEration\",\n",
    "    \"DATATOP\": \"DATATOP\",\n",
    "    \"SZTE\": \"SZTE\",\n",
    "    \"TRD\": \"Trondheim Study\",\n",
    "    \"MDGAP-SW\": \"MD GAP - South West Dementia Brain Bank - Southmead Hospital\",\n",
    "    \"S4\": \"S4\",\n",
    "    \"NETPDLS1\": \"NET-PD\",\n",
    "    \"DENOPA\": \"DeNoPa\",\n",
    "    \"NTUH\": \"Genomewide Association Study in PD among East Asians - NTUH\",\n",
    "    \"BRFAA\": \"Biological markers in Parkinson's disease - BRFAA\",\n",
    "    \"BUE\": \"BUE - GP2\",\n",
    "    \"LARGEPD\": \"To be updated - Cleveland Clinic\",\n",
    "    \"BBDP\": \"Arizona Brain Bank Brain and body donation programme (BBDP)\",\n",
    "    \"CINCINNATI\": \"Clinical and genetic study of Unusual/Familial Movement Disorders\",\n",
    "    \"MMCTU\": \"Genotype-Phenotype Correlations in Nepalese PD Population - MMCTU\",\n",
    "    \"PERUGIA\": \"CSF Biomarkers\",\n",
    "    \"STELLENBOS\": \"Genetic study of PD and related movement disorders\",\n",
    "    \"CHULALONGK\": \"Genetic Testing in Patients with Movement Disorders and PD\",\n",
    "    \"MDGAP-NWCSTL\": \"MD GAP - Newcastle Brain Bank - University of Newcastle\",\n",
    "    \"KFSH\": \"The Molecular Basis of Movement Disorders in Saudis - KFSH\",\n",
    "    \"PAGE\": \"PAGE\",\n",
    "    \"SYNAPS-GR\": \"SYNAPS-GR\",\n",
    "    \"PDCARE\": \"PDCARE\",\n",
    "    \"RAPSODI\": \"RAPSODI\",\n",
    "    \"YOPDFL\": \"YOPDFL (Young Onset Parkinson's Disease)\",\n",
    "    \"FDSANTIAGO\": \"GP2 - Fundación Diagnosis\",\n",
    "    \"MULODZ\": \"Molecular Testing of Polish PD Patients\",\n",
    "    \"TONIC-PERRON\": \"TONiC PD\",\n",
    "    \"KOC\": \"GP2 - Koc University\",\n",
    "    \"JGUM\": \"JGUM\",\n",
    "    \"PATRAS\": \"PATRAS\",\n",
    "    \"NUGENE\": \"NU Gene\",\n",
    "    \"CDPDGR\": \"Thessaly PD Database\",\n",
    "    \"JENPARK\": \"Jena Parkinson Study\",\n",
    "    \"UROMETV\": \"Standardized data collection of PD patients\",\n",
    "    \"CHCHPD\": \"Christchurch Longitudinal Parkinson's Study\",\n",
    "    \"HIPDR\": \"HIPDR\",\n",
    "    \"KOLLING\": \"Kolling PD\",\n",
    "    \"SEOULNUH\": \"GP2 - Seoul National University Hospital\",\n",
    "    \"NROSF2U\": \"Multicentric Retrospective and Prospective PD Database\",\n",
    "    \"BLAACPD-LSU\": \"BLAACPD - Louisiana State University\",\n",
    "    \"IMMUNEPD\": \"Innate and Adaptive Immunity in PD\",\n",
    "    \"PROMO-EPIPARK\": \"PROMO-EPIPARK\",\n",
    "    \"PROMO-GENEPARK\": \"PROMO-GENEPARK\",\n",
    "    \"PROMO-ROPAD\": \"PROMO-ROPAD\",\n",
    "    \"PROMO-SFB134\": \"PROMO-SFB134\",\n",
    "    \"PROMO-SYSMED\": \"PROMO-SYSMEDPD\",\n",
    "    \"OXQUIP\": \"Oxford Study of Quantification in Parkinsonism\",\n",
    "    \"PROMO-USTP\": \"PROMO-USTP\",\n",
    "    \"MUVIENNA\": \"Families with Neurodegenerative Diseases\",\n",
    "    \"CATANZARO\": \"Familial and Early-onset PD Database\",\n",
    "    \"QMAP\": \"Quantitative MRI for Anatomical Phenotyping in Parkinson's Disease\",\n",
    "    \"CATPD\": \"Central Asian and Transcaucasian PD (CAT-PD)\",\n",
    "    \"IPDGCAF-NG\": \"IPDGC Africa - The Nigerian Parkinson Disease Research Network\",\n",
    "    \"IPDGCEA-JU\": \"IPDGC East Asia - Juntendo University\",\n",
    "    \"TUMBB\": \"Biobank for Diseases of the Peripheral and Central Nervous System - TUMBB\",\n",
    "    \"TUMPD\": \"Genetics of Parkinson's Disease - TUMPD\",\n",
    "    \"ANANEOS\": \"ANANEOS Study\",\n",
    "    \"MDGAP-CAMBRIDGE\": \"MD GAP - Cambridge\",\n",
    "    \"MDGAP-OXF\": \"MD GAP - OXF\",\n",
    "    \"VETPD\": \"VET-PD\",\n",
    "    \"BERNGEN\": 'BERN',\n",
    "    \"SINAI\": 'Mount Sinai Cohort',\n",
    "    \"PARKWEST\": \"The Norwegian Parkwest Study\",\n",
    "    \"PROSPECT\": \"PROSPECT\",\n",
    "    \"UMUDALL\": \"UMUDALL\",\n",
    "    \"TORONTO\": \"To be updated - University of Toronto\",\n",
    "    \"MDGAP-VBB\": \"MD GAP - Victorian Brain Bank\",\n",
    "    \"IRCCSN\": \"Genetics of Neurological Pathologies: New Variations, New Phenotypes\",\n",
    "    \"PROBEPD\": \"PROBEPD\",\n",
    "    \"TLVMC\": \"DLB Cohort TLVMC\",\n",
    "    \"ANDPD\": \"ANDPD\",    \n",
    "    \"MONPAR\": \"Genetic Study for Pakinson's Disease in Mongolia\",\n",
    "    \"NGIPDO\": \"NGIPDO - Neurogenomics and Informatics Center - Parkinson's Disease  Deep Molecular Profiling\",\n",
    "    \"IPDGCEA-UKM\": \"IPDGC East Asia - UKM\",\n",
    "    \"DODONA\": \"The DodoNa Project: DNA Predictions to Improve Neurological Health\",\n",
    "    \"PROMO-CARRIERS\": \"PROMO-CARRIERS\",\n",
    "    \"HAWAIIPD\": \"HAWAII PD\",\n",
    "    \"YMS\": \"Yale Medical School\",\n",
    "    \"IGECND\": \"IGECND\",\n",
    "    \"DHAKA\": \"GP2 - Bangladesh\",\n",
    "    \"UGCPD\": \"UGCPD\",\n",
    "    \"AAPDGC\": \"AfrAbia-PD Genomic Consortium (AA-PDGC)\",\n",
    "    \"REGARDS\": \"REasons for Geographic and Racial Differences in Stroke (REGARDS) Project\",\n",
    "    \"NZPEGS\": \"New Zealand Parkinson's Environment and Genes Study\",\n",
    "    \"GPX\": \"GenePool Study\",\n",
    "    \"VAIBB\": \"Van Andel Institute Brain Bank\",\n",
    "    \"WEHI\": \"WEHI - Integrating molecular and omics approaches in Parkinson’s disease\",\n",
    "    \"ACCESSPD\": \"ACCESSPD\",\n",
    "    \"GPS\": \"GenePool Study\",\n",
    "    \"NGP\": \"The Neuro Genomics Partnership\",\n",
    "    \"SCANPSP\": \"SCANPSP - Study of Comprehensive ANd Multimodal Marker-based Cohort of PSP\",\n",
    "    \"SAMSUNG\": \"GP2 - Samsung\",\n",
    "    \"BURGOS\": \"Local Study - Burgos\",    \n",
    "    \"SP-DON\": None,\n",
    "    \"ECMPDC\": None,\n",
    "    \"SP-GRANADA\": None,\n",
    "    \"AHEPA\": None,\n",
    "    \"CIDR\": None,\n",
    "    \"INKOSI\": None,\n",
    "    \"SP-BCN1\": None,\n",
    "    \"SANGERARDO\": None,\n",
    "    \"GEMELLI\": None,\n",
    "    \"NMDAT\": None,\n",
    "    \"TARTU\": None,\n",
    "    \"TEHRAN\": None,\n",
    "    \"KIEL\": None,\n",
    "    \"YAGUE\": None,\n",
    "    \"MAYO\": None,\n",
    "    \"AMSBB\": None,\n",
    "    \"PDSA\": None,\n",
    "    \"RDIPD\": None,\n",
    "    \"ANKARA\": None,\n",
    "    \"DUPARC\": None,\n",
    "    \"ESTBB\": None,\n",
    "    \"PUJBC\": None,\n",
    "    \"CMC\": None,\n",
    "    \"SAREMA\": None,\n",
    "    \"IPDGCEA-MU\": None,\n",
    "    \"AARHUS\": None,\n",
    "    \"CEGEMOD\": None,\n",
    "    \"JUNT\": None,\n",
    "    \"HSPH\": None,\n",
    "    \"NGRC\": None,\n",
    "    \"SILESIA\": None,\n",
    "    \"KRAKOW\": None,\n",
    "    \"BRNO\": None,\n",
    "    \"COLOMBO\": None,\n",
    "    \"ADRC\": None,\n",
    "    \"SILAD\": None,\n",
    "    \"STANPD\": None,\n",
    "    \"GENNIS\": None,\n",
    "    \"AHS\": None,\n",
    "    \"IDIBAPS\": None,\n",
    "    \"FLENI\": None,\n",
    "    \"EPIC\": None,\n",
    "    \"CCFPD\": None,\n",
    "    \"GRACE\": None,\n",
    "    \"NYSCF\": None,\n",
    "    \"PARS\": None,\n",
    "    \"PEG\": None,\n",
    "    \"SAMC\": None,\n",
    "    \"SAUD\": None,\n",
    "    \"COBB\": None,\n",
    "    \"PDCGC\": None,\n",
    "    \"SANTPAU\": None,\n",
    "    \"TRAPCAF\": None,\n",
    "    \"NITHAI\": None,\n",
    "    \"RTGRS\": None,\n",
    "    \"PROGENI\": None,\n",
    "    \"BIOPD\": None,\n",
    "    \"BOLOGNA\": None,\n",
    "    \"BPS\": None,\n",
    "    \"CABBS\": None,\n",
    "    \"CLGN\": None,\n",
    "    \"CUHK\": None,\n",
    "    \"GENCI\": None,\n",
    "    \"HEBA-BRC\": None,\n",
    "    \"HEBA-INNS\": None,\n",
    "    \"HEBA-KASSEL\": None,\n",
    "    \"INECO\": None,\n",
    "    \"INNS\": None,\n",
    "    \"MONT\": None,\n",
    "    \"NORTHSHORE\": None,\n",
    "    \"PARAGEN\": None,\n",
    "    \"FAIRPARK\": None,\n",
    "    \"PROLBD\": None,\n",
    "    \"RIMU\": None,\n",
    "    \"PARAGEN-MON\": None,\n",
    "    \"UCIPARK\": None,\n",
    "    \"GBA1NL\": None,\n",
    "    \"PDNEF\": None,\n",
    "    \"UCIPROS\": None,\n",
    "    \"NHGRI\": None,\n",
    "    \"CAGL\": None,\n",
    "    \"REMANT\": None,\n",
    "    \"REMEL\": None,\n",
    "    \"UDINE\": None,\n",
    "    \"REMEDIO\": None,\n",
    "    \"PAIR-PSP\": None,\n",
    "    \"PDBIOMARKERS\": None,\n",
    "    \"NAFPD\": None,\n",
    "    \"FNPREG\": None,\n",
    "    \"GED\": None,\n",
    "    \"PDGEN-CLEM\": None,\n",
    "    \"MDGAP-DEMENTIA\": None,\n",
    "    \"EXENATIDEPD3\": None,\n",
    "    \"GPDC\": None,\n",
    "    \"NORBANK\": None,\n",
    "    \"NYBB\": None,\n",
    "    \"NYPUM/PARKNY\": None,\n",
    "    \"UWBRAIN\": None,\n",
    "    \"SUREPD2\": None,\n",
    "    \"UNIBS\": None,\n",
    "    \"IUC\": None,\n",
    "    \"CANDAS-SMPD\": None,\n",
    "    \"CANDAS-PDGEN\": None,\n",
    "    \"EMORY\": None,\n",
    "    \"CANPARK\": None,\n",
    "    \"TAS\": None,\n",
    "    \"ISLAND\": None,\n",
    "    \"PPRBD\": None,\n",
    "    \"CRUCES\": None,\n",
    "    \"NRIBBB\": None,\n",
    "    \"NBB-HVD\": None,\n",
    "    \"NBB-SINAI\": None,\n",
    "    \"NBB-SPLVD\": None,\n",
    "    \"NBB-MLND\": None,\n",
    "    \"NBB-MIAMI\": None,\n",
    "    \"NBB-PBG\": None,\n",
    "    \"BELGRADE\": None,\n",
    "    \"MHHBB\": None,\n",
    "    \"SLOWSPEED\": None,\n",
    "    \"FMGC\": None,\n",
    "    \"IRAMS\": None,\n",
    "    \"GENOT\": None,\n",
    "    \"UTSW\": None,\n",
    "    \"ADPAR\": None,\n",
    "    \"NIMH\": None\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_map_df = pd.DataFrame(list(cohort_mapping.items()), columns=['gt', 'gmmb'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "tracker_merged = cohort_map_df.merge(gmmb_df, how='outer', left_on='gmmb', right_on='Cohort Name')\n",
    "tracker_merged.rename(columns={'gt':'genotracker'}, inplace=True)\n",
    "tracker_merged.drop(columns=['gmmb'], inplace=True)\n",
    "tracker_merged.to_csv('data/cohort_dashboard_updated.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## add data to database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "df_sub = df.loc[:,[\n",
    "    \"Study Code\",\n",
    "    \"Monogenic / Complex / Mixed\",\n",
    "    \"City\",\n",
    "    \"Geographic locality [free text]\",\n",
    "    \"DNA samples attempted [N]\",\n",
    "    \"Total samples passing basic QC [N]\",\n",
    "    \"Failed call rates [N < 95%]\",\n",
    "    \"Failed sex check [N with contradictory reported versus genetic sex]\",\n",
    "    \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\",\n",
    "    \"Duplicate samples [N, IBD estiamtes > 95%]\",\n",
    "    \"Imputation panels(s) [Free text]\",\n",
    "    \"Imputation complete for current samples [0 = no, 1 = yes]\",\n",
    "    \"Full data [0 = only summary stats, 1 = genotypes sharable]\",\n",
    "    \"GDPR [0 = no European restriction, 1 = limited by GDPR]\",\n",
    "    \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\"\n",
    "]]\n",
    "\n",
    "rename_dict = {\n",
    "    \"Study Code\": \"study_code\",\n",
    "    \"Monogenic / Complex / Mixed\": \"monogenic_complex_mixed\",\n",
    "    \"City\": \"city\",\n",
    "    \"Geographic locality [free text]\": \"geographic_locality\",\n",
    "    \"DNA samples attempted [N]\": \"n_dna_samples_attempted\",\n",
    "    \"Total samples passing basic QC [N]\": \"total_qc_pass\",\n",
    "    \"Failed call rates [N < 95%]\": \"callrate_fails\",\n",
    "    \"Failed sex check [N with contradictory reported versus genetic sex]\": \"sex_fails\",\n",
    "    \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\": \"het_fails\",\n",
    "    \"Duplicate samples [N, IBD estiamtes > 95%]\": \"duplicates\",\n",
    "    \"Imputation panels(s) [Free text]\": \"imputation_panel\",\n",
    "    \"Imputation complete for current samples [0 = no, 1 = yes]\": \"imputation_complete\",\n",
    "    \"Full data [0 = only summary stats, 1 = genotypes sharable]\": \"genotypes_shareable\",\n",
    "    \"GDPR [0 = no European restriction, 1 = limited by GDPR]\": \"gdpr\",\n",
    "    \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\": \"site\"\n",
    "}\n",
    "\n",
    "df_sub.rename(columns=rename_dict, inplace=True)\n",
    "\n",
    "df_out = df_sub.fillna({\n",
    "    'study_code': '',\n",
    "    'monogenic_complex_mixed': '',\n",
    "    'city': '',\n",
    "    'geographic_locality': '',\n",
    "    'n_dna_samples_attempted': 0,\n",
    "    'total_qc_pass': 0,\n",
    "    'callrate_fails': 0,\n",
    "    'sex_fails': 0,\n",
    "    'het_fails': 0,\n",
    "    'duplicates': 0,\n",
    "    'genotypes_shareable': 0,\n",
    "    'imputation_complete': 0,\n",
    "    'imputation_panel': '',\n",
    "    'gdpr': 0,\n",
    "    'site': ''\n",
    "})\n",
    "\n",
    "numeric_columns = [\n",
    "    'n_dna_samples_attempted', 'total_qc_pass', 'callrate_fails', 'sex_fails', \n",
    "    'het_fails', 'duplicates', 'genotypes_shareable', 'imputation_complete', 'gdpr'\n",
    "]\n",
    "\n",
    "df_out[numeric_columns] = df_out[numeric_columns].astype(int)\n",
    "df_out.to_csv('data/genotracker_clean.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_csv('data/GenoTracker.csv', header=1)\n",
    "# df_sub = df.loc[:,[\n",
    "#     \"Study Code\",\n",
    "#     \"Monogenic / Complex / Mixed\",\n",
    "#     \"City\",\n",
    "#     \"Geographic locality [free text]\",\n",
    "#     \"DNA samples attempted [N]\",\n",
    "#     \"Total samples passing basic QC [N]\",\n",
    "#     \"Failed call rates [N < 95%]\",\n",
    "#     \"Failed sex check [N with contradictory reported versus genetic sex]\",\n",
    "#     \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\",\n",
    "#     \"Duplicate samples [N, IBD estiamtes > 95%]\",\n",
    "#     \"African (AFR)\",\n",
    "#     \"African admixed (AAC)\",\n",
    "#     \"Ashkenzai (AJ)\",\n",
    "#     \"East Asian (EAS)\",\n",
    "#     \"European (EUR)\",\n",
    "#     \"Finnish (FIN)\",\n",
    "#     \"LatinX (AMR)\",\n",
    "#     \"South Asian (SAS)\",\n",
    "#     \"Central Asian (CAS)\",\n",
    "#     \"Middle Eastern (MDE)\",\n",
    "#     \"Complex Admixture (CAH)\",\n",
    "#     \"Total\",\n",
    "#     \"Imputation panels(s) [Free text]\",\n",
    "#     \"Imputation complete for current samples [0 = no, 1 = yes]\",\n",
    "#     \"Full data [0 = only summary stats, 1 = genotypes sharable]\",\n",
    "#     \"GDPR [0 = no European restriction, 1 = limited by GDPR]\",\n",
    "#     \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\"\n",
    "# ]]\n",
    "\n",
    "# rename_dict = {\n",
    "#     \"Study Code\": \"study_code\",\n",
    "#     \"Monogenic / Complex / Mixed\": \"monogenic_complex_mixed\",\n",
    "#     \"City\": \"city\",\n",
    "#     \"Geographic locality [free text]\": \"geographic_locality\",\n",
    "#     \"DNA samples attempted [N]\": \"n_dna_samples_attempted\",\n",
    "#     \"Total samples passing basic QC [N]\": \"total_qc_pass\",\n",
    "#     \"Failed call rates [N < 95%]\": \"callrate_fails\",\n",
    "#     \"Failed sex check [N with contradictory reported versus genetic sex]\": \"sex_fails\",\n",
    "#     \"Failed contamination and heterozygosity checks [N, outliers for ancestry specific heterozygosity rates]\": \"het_fails\",\n",
    "#     \"Duplicate samples [N, IBD estiamtes > 95%]\": \"duplicates\",\n",
    "#     \"African (AFR)\": \"AFR\",\n",
    "#     \"African admixed (AAC)\": \"AAC\",\n",
    "#     \"Ashkenzai (AJ)\": \"AJ\",\n",
    "#     \"East Asian (EAS)\": \"EAS\",\n",
    "#     \"European (EUR)\": \"EUR\",\n",
    "#     \"Finnish (FIN)\": \"FIN\",\n",
    "#     \"LatinX (AMR)\": \"AMR\",\n",
    "#     \"South Asian (SAS)\": \"SAS\",\n",
    "#     \"Central Asian (CAS)\": \"CAS\",\n",
    "#     \"Middle Eastern (MDE)\": \"MDE\",\n",
    "#     \"Complex Admixture (CAH)\": \"CAH\",\n",
    "#     \"Total\": \"total\",\n",
    "#     \"Imputation panels(s) [Free text]\": \"imputation_panel\",\n",
    "#     \"Imputation complete for current samples [0 = no, 1 = yes]\": \"imputation_complete\",\n",
    "#     \"Full data [0 = only summary stats, 1 = genotypes sharable]\": \"genotypes_shareable\",\n",
    "#     \"GDPR [0 = no European restriction, 1 = limited by GDPR]\": \"gdpr\",\n",
    "#     \"Site [NIH, Tubingen/DZNE, Fulgent, UCL]\": \"site\"\n",
    "# }\n",
    "\n",
    "# df_sub.rename(columns=rename_dict, inplace=True)\n",
    "\n",
    "# df_out = df_sub.fillna({\n",
    "#     'study_code': '',\n",
    "#     'monogenic_complex_mixed': '',\n",
    "#     'city': '',\n",
    "#     'geographic_locality': '',\n",
    "#     'n_dna_samples_attempted': 0,\n",
    "#     'total_qc_pass': 0,\n",
    "#     'callrate_fails': 0,\n",
    "#     'sex_fails': 0,\n",
    "#     'het_fails': 0,\n",
    "#     'duplicates': 0,\n",
    "#     'AFR': 0,\n",
    "#     'AAC': 0,\n",
    "#     'AJ': 0,\n",
    "#     'EAS': 0,\n",
    "#     'EUR': 0,\n",
    "#     'FIN': 0,\n",
    "#     'AMR': 0,\n",
    "#     'SAS': 0,\n",
    "#     'CAS': 0,\n",
    "#     'MDE': 0,\n",
    "#     'CAH': 0,\n",
    "#     'total': 0,\n",
    "#     'genotypes_shareable': 0,\n",
    "#     'imputation_complete': 0,\n",
    "#     'imputation_panel': '',\n",
    "#     'gdpr': 0,\n",
    "#     'site': ''\n",
    "# })\n",
    "\n",
    "# numeric_columns = [\n",
    "#     'n_dna_samples_attempted', 'total_qc_pass', 'callrate_fails', 'sex_fails', \n",
    "#     'het_fails', 'duplicates', 'AFR', 'AAC', 'AJ', 'EAS', 'EUR', 'FIN', \n",
    "#     'AMR', 'SAS', 'CAS', 'MDE', 'CAH', 'total', 'genotypes_shareable', \n",
    "#     'imputation_complete', 'gdpr'\n",
    "# ]\n",
    "\n",
    "# df_out[numeric_columns] = df_out[numeric_columns].astype(int)\n",
    "# df_out.to_csv('data/genotracker_clean.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry_df = pd.read_csv('data/FEB_2024_cohort_ancestry_phenotype.csv')\n",
    "ancestry_df.rename(columns={'Cohort':'study_code'}, inplace=True)\n",
    "df_ancestry_out = df_out.merge(ancestry_df, on='study_code', how='left')\n",
    "columns_to_rename = list(ancestry_df.columns[1:])\n",
    "rename_mapping = {col: col.lower().replace(' ', '_') for col in columns_to_rename}\n",
    "df_ancestry_out.rename(columns=rename_mapping, inplace=True)\n",
    "df_ancestry_out.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "for col in rename_mapping.values():\n",
    "    df_ancestry_out[col] = df_ancestry_out[col].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ancestry_out.to_csv('data/genotracker_clean.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['study_code', 'monogenic_complex_mixed', 'city', 'geographic_locality',\n",
       "       'n_dna_samples_attempted', 'total_qc_pass', 'callrate_fails',\n",
       "       'sex_fails', 'het_fails', 'duplicates', 'imputation_panel',\n",
       "       'imputation_complete', 'genotypes_shareable', 'gdpr', 'site',\n",
       "       'afr_case', 'afr_control', 'afr_other', 'aac_case', 'aac_control',\n",
       "       'aac_other', 'aj_case', 'aj_control', 'aj_other', 'eas_case',\n",
       "       'eas_control', 'eas_other', 'eur_case', 'eur_control', 'eur_other',\n",
       "       'fin_case', 'fin_control', 'fin_other', 'amr_case', 'amr_control',\n",
       "       'amr_other', 'sas_case', 'sas_control', 'sas_other', 'cas_case',\n",
       "       'cas_control', 'cas_other', 'mde_case', 'mde_control', 'mde_other',\n",
       "       'cah_case', 'cah_control', 'cah_other', 'total'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ancestry_out.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   id study_code monogenic_complex_mixed           city geographic_locality  \\\n",
      "0   1        BCM                 Complex     Houston,TX                 USA   \n",
      "1   2    CORIELL                 Complex     Camden, NJ                 USA   \n",
      "2   3    PDGNRTN                   Mixed     Miami, FL                  USA   \n",
      "3   4        UMD                 Complex  Baltimore, MD                 USA   \n",
      "4   5    PROSPOS                 Complex           Oslo              Norway   \n",
      "\n",
      "   n_dna_samples_attempted  total_qc_pass  callrate_fails  sex_fails  \\\n",
      "0                      809            788               7         12   \n",
      "1                     8982           8718             140         56   \n",
      "2                     4694           3791             239        352   \n",
      "3                      479            441               6         15   \n",
      "4                       96             95               0          1   \n",
      "\n",
      "   het_fails  ...  CAS  MDE  CAH  total  genotyping_complete  \\\n",
      "0          0  ...    5    4    5    788                    0   \n",
      "1          1  ...    8   48  167   8718                    0   \n",
      "2          0  ...   16   21  174   3791                    0   \n",
      "3          0  ...    2    2    2    441                    0   \n",
      "4          0  ...    0    0    0     95                    0   \n",
      "\n",
      "   imputation_panel  imputation_complete  genotypes_shareable  gdpr     site  \n",
      "0            TOPMED                    1                    1     0      NIH  \n",
      "1            TOPMED                    1                    1     0      NIH  \n",
      "2            TOPMED                    1                    1     0  Fulgent  \n",
      "3            TOPMED                    1                    1     0      NIH  \n",
      "4            TOPMED                    1                    1     1  Fulgent  \n",
      "\n",
      "[5 rows x 29 columns]\n",
      "               id  n_dna_samples_attempted  total_qc_pass  callrate_fails  \\\n",
      "count  311.000000               311.000000     311.000000      311.000000   \n",
      "mean   156.000000               189.302251     175.440514        5.173633   \n",
      "std     89.922189               761.131042     716.178651       24.136081   \n",
      "min      1.000000                 0.000000       0.000000        0.000000   \n",
      "25%     78.500000                 0.000000       0.000000        0.000000   \n",
      "50%    156.000000                 0.000000       0.000000        0.000000   \n",
      "75%    233.500000                19.000000      18.500000        0.000000   \n",
      "max    311.000000              8982.000000    8718.000000      239.000000   \n",
      "\n",
      "        sex_fails   het_fails  duplicates          AFR         AAC  \\\n",
      "count  311.000000  311.000000  311.000000   311.000000  311.000000   \n",
      "mean     4.427653    0.009646    3.559486     8.501608    3.578778   \n",
      "std     22.928650    0.097898   20.679010    91.034339   41.613731   \n",
      "min      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "25%      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "50%      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "75%      0.000000    0.000000    0.000000     0.000000    0.000000   \n",
      "max    352.000000    1.000000  296.000000  1403.000000  723.000000   \n",
      "\n",
      "               AJ  ...         AMR         SAS         CAS         MDE  \\\n",
      "count  311.000000  ...  311.000000  311.000000  311.000000  311.000000   \n",
      "mean     8.562701  ...    2.099678    2.073955    2.906752    1.871383   \n",
      "std     62.169345  ...   15.717251   12.055423   28.430918   12.646159   \n",
      "min      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "25%      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "50%      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "75%      0.000000  ...    0.000000    0.000000    0.000000    0.000000   \n",
      "max    780.000000  ...  195.000000  142.000000  366.000000  197.000000   \n",
      "\n",
      "              CAH        total  genotyping_complete  imputation_complete  \\\n",
      "count  311.000000   311.000000                311.0           311.000000   \n",
      "mean     2.742765   175.440514                  0.0             0.257235   \n",
      "std     22.141293   716.178651                  0.0             0.437814   \n",
      "min      0.000000     0.000000                  0.0             0.000000   \n",
      "25%      0.000000     0.000000                  0.0             0.000000   \n",
      "50%      0.000000     0.000000                  0.0             0.000000   \n",
      "75%      0.000000    18.500000                  0.0             1.000000   \n",
      "max    299.000000  8718.000000                  0.0             1.000000   \n",
      "\n",
      "       genotypes_shareable        gdpr  \n",
      "count           311.000000  311.000000  \n",
      "mean              0.334405    0.202572  \n",
      "std               0.472542    0.402564  \n",
      "min               0.000000    0.000000  \n",
      "25%               0.000000    0.000000  \n",
      "50%               0.000000    0.000000  \n",
      "75%               1.000000    0.000000  \n",
      "max               1.000000    1.000000  \n",
      "\n",
      "[8 rows x 23 columns]\n",
      "id                          int64\n",
      "study_code                 object\n",
      "monogenic_complex_mixed    object\n",
      "city                       object\n",
      "geographic_locality        object\n",
      "n_dna_samples_attempted     int64\n",
      "total_qc_pass               int64\n",
      "callrate_fails              int64\n",
      "sex_fails                   int64\n",
      "het_fails                   int64\n",
      "duplicates                  int64\n",
      "AFR                         int64\n",
      "AAC                         int64\n",
      "AJ                          int64\n",
      "EAS                         int64\n",
      "EUR                         int64\n",
      "FIN                         int64\n",
      "AMR                         int64\n",
      "SAS                         int64\n",
      "CAS                         int64\n",
      "MDE                         int64\n",
      "CAH                         int64\n",
      "total                       int64\n",
      "genotyping_complete         int64\n",
      "imputation_panel           object\n",
      "imputation_complete         int64\n",
      "genotypes_shareable         int64\n",
      "gdpr                        int64\n",
      "site                       object\n",
      "dtype: object\n",
      "Empty DataFrame\n",
      "Columns: [id, study_code, monogenic_complex_mixed, city, geographic_locality, n_dna_samples_attempted, total_qc_pass, callrate_fails, sex_fails, het_fails, duplicates, AFR, AAC, AJ, EAS, EUR, FIN, AMR, SAS, CAS, MDE, CAH, total, genotyping_complete, imputation_panel, imputation_complete, genotypes_shareable, gdpr, site]\n",
      "Index: []\n",
      "\n",
      "[0 rows x 29 columns]\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Create an engine instance\n",
    "engine = create_engine('sqlite:///test.db')\n",
    "\n",
    "# Query to select all data from the genetic_data table\n",
    "query = \"SELECT * FROM genetic_data\"\n",
    "\n",
    "# Execute the query and load the data into a pandas DataFrame\n",
    "df = pd.read_sql(query, engine)\n",
    "\n",
    "# Display the first few rows of the DataFrame\n",
    "print(df.head())\n",
    "\n",
    "# Display summary statistics\n",
    "print(df.describe())\n",
    "\n",
    "# Display the data types of the columns\n",
    "print(df.dtypes)\n",
    "\n",
    "# Filter the data\n",
    "df_filtered = df[df['city'] == 'Boulder']\n",
    "print(df_filtered.head())\n",
    "\n",
    "# # Plot the data\n",
    "# import matplotlib.pyplot as plt\n",
    "\n",
    "# df['n_dna_samples_attempted'].hist()\n",
    "# plt.xlabel('Number of DNA Samples Attempted')\n",
    "# plt.ylabel('Frequency')\n",
    "# plt.title('Distribution of DNA Samples Attempted')\n",
    "# plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
